Création du tutoriel (#1)
DoryanR/proposition-nouveau-tutoriel-de-contribution-osu-french: Comment contribuer au projet de traduction de l'osu! wiki.md

[b]Une fois le dossier ouvert, j’aimerais vous parler de quelques règles et vous donner des conseils concernant la traduction, afin que vous réalisez cette dernière correctement :[/b]

[list]
[*]GitHub à mis en place un système de [b]branche[/b] (branch en anglais), qui vous permet d’inclure des modifications sur plusieurs de ces branches, ce qui vous permet de traduire plusieurs fichiers différents, sans que cela puisse vous gêner avec la précédente.
Pour créer une branche, vous devez aller dans GitHub Desktop, et cliquer sur [b][url=https://i.imgur.com/t7i3OFZ.png]ce bouton[/url][/b], qui va vous ouvrir [b][url=https://i.imgur.com/zvgusd1.png]une liste déroulante comme celle-ci[/url][/b] avec toutes les branches que vous possédez (vous n’en aurez aucune au départ).

Pour en créer une, cliquer sur [b]New branch[/b] en haut à droite, mettez ensuite le nom de votre branche (nous vous conseillons de mettre le nom de l’article que vous souhaitez de traduire (en remplaçant les espaces par des [b]-[/b] comme le nom ne doit pas contenir d’espaces, sinon il le fera pour vous), avec juste après la mention [b]FR[/b] pour signifier que c’est une traduction en langue française.

> Quand vous aurez renommer votre branche, cliquer sur le bouton [b]Create branch[/b], et le tour est joué !

Author: ```suggestion
Lorsque vous aurez nommé votre branche, cliquez sur le bouton [b]Create branch[/b], et le tour est joué !
```

C'est mieux comme ça, car la c'est pas une procédure pour renommer une branche, mais pour en créer une

En tant que votre premier commit, vous n’avez pas de [b]Pull Request[/b] d’actif, vous devez donc en créer une.
Une Pull Request (PR en abrégé), c’est l’endroit où vous allez proposer votre modification sur le projet, pour qu’il soit critiquer, et qu’on vous propose des modifications, le cas échéant on va vous accepter vos modifications si nous n’avons rien à redire.

Pour en créer une, on va rester sur GitHub Desktop, et remarquer que la fenêtre d’origine à [b][url=https://i.imgur.com/JoQKxOC.png]légèrement changer[/url][/b].
Vous avez ici sur ce screen en bleu un message qui vous dit ce qui a été dit plus haut : que votre branche n’est pas publiée sur GitHub, et n’a donc pas de Pull Request à son actif. Pour la créer, vous pouvez cliquer sur [b]Publish Branch[/b] à droite du message, ou cliquer en haut sur [b]Publish Branch[/b], juste à côté d’où vous avez créé votre branche. Les deux actions amènent au même résultat.
Une fois cliquer sur l’un des deux boutons, votre branche va être publier (on appelle ça aussi [b]push[/b]) sur GitHub.

> Désormais, vous pouvez revenir sur [b][url=https://github.com/ppy/osu-wiki]l’accueil du dépôt[/url][/b], ou rester sur GitHub Desktop, et cliquer sur [b]Create Pull Request[/b] au centre de la fenêtre, au même endroit ou on nous proposait de publier la branche sur GitHub.

Reviewer: ```suggestion
Désormais, vous pouvez revenir sur [b][url=https://github.com/ppy/osu-wiki]l’accueil du dépôt[/url][/b], ou rester sur GitHub Desktop et cliquer sur [b]Create Pull Request[/b] au centre de la fenêtre, au même endroit où il vous propose de publier la branche sur GitHub.
```

> Si vous choisissez de retourner sur le dépôt, vous allez avoir un [b][url=https://i.imgur.com/1sSHcQZ.png]petit changement[/url][/b] en haut de votre écran. Cette fenêtre en orange est la branche que vous venez juste de publier sur GitHub, cliquez donc sur [b]Compare & pull request[/b].

Reviewer: ```suggestion
Si vous choisissez de retourner sur le dépôt, vous pourrez observer un [b][url=https://i.imgur.com/1sSHcQZ.png]petit changement[/url][/b] en haut de votre écran. Cette fenêtre en orange est la branche que vous venez juste de publier sur GitHub, cliquez donc sur [b]Compare & pull request[/b].
```

> Vous allez arriver sur [b][url=https://i.imgur.com/9ENAr8z.png]cette fenêtre[/url][/b]. C’est là ou vous avez les informations sur votre branche, avec les modifications que vous voulez proposer.

Reviewer: ```suggestion
Vous allez arriver sur [b][url=https://i.imgur.com/9ENAr8z.png]cette fenêtre[/url][/b]. C’est là que vous pouvez voir les informations sur votre branche, avec les modifications que vous voulez proposer.
```

> Ce qui se trouve tout en haut, ne vous en occupez pas, ce sont juste les paramètres qui définissent sur quel dépôt du projet doit il se baser pour qu’il soit publié par la suite dessus. De base c’est bien configurer, il n’y a pas lieu d’y changer.

Reviewer: ```suggestion
Ce qui se trouve tout en haut, ne vous en occupez pas, ce sont simplement les paramètres qui définissent sur quel dépôt du projet il doit se baser pour qu’il soit publié par la suite.
```

> Juste en dessous, vous avez le titre de votre premier commit, qui est automatiquement transformé en votre titre pour votre Pull Request, et avec la description en dessous (si vous en avez ajouté une).

Reviewer: ```suggestion
Juste en dessous, vous avez le titre de votre premier commit, qui est automatiquement transformé en votre titre pour votre Pull Request, avec la description en dessous (si vous en avez ajouté une).
```

Si vous allez en dessous, vous avez le même affichage que tout à l’heure sur GitHub Desktop, avec ce que vous avez ajouté au fichier (ou supprimé).

> Une petite information importante : en bas de la zone de texte pour la description, vous avez [b][url=https://i.imgur.com/1ucKWCA.png]ceci[/url][/b]. Ce paramètre est à laisser cocher, car il permet aux modérateurs du projet (ils sont appelés des [b]Maintainers[/b]) d’accéder à votre branche et d’y apporter des modifications, comme ce que vous ferez plus tard. Cela n’est pas gênant de les laisser l’accès à votre branche, car seul eux pourront y accéder, et ils sont de confiance, n’ayez crainte !

Reviewer: ```suggestion
Une petite information importante : en bas de la zone de texte pour la description, vous avez [b][url=https://i.imgur.com/1ucKWCA.png]ceci[/url][/b]. Ce paramètre doit être coché, car il permet aux modérateurs du projet (ils sont appelés [b]Maintainers[/b]) d’accéder à votre branche et d’y apporter des modifications, comme ce que vous ferez plus tard. Cela n’est pas gênant de leur laisser l’accès à votre branche, car seuls eux pourront y accéder.
```

> Quand vous aurez ajouté une description (vous pouvez mettre que vous avez traduit tel article, cela suffira), cliquer sur [b]Create pull request[/b].

Reviewer: ```suggestion
Lorsque vous aurez ajouté une description (vous pouvez simplement dire que vous avez traduit tel article, cela suffira), cliquez sur [b]Create pull request[/b].
```

> Une fois votre Pull Request publié, voici [b][url=https://i.imgur.com/VsIAK7L.png]à quoi cela ressemble[/url][/b].

Reviewer: ```suggestion
Une fois votre Pull Request publiée, voici [b][url=https://i.imgur.com/VsIAK7L.png]à quoi cela ressemble[/url][/b].
```

> A partir de ce moment, vous avez fait le plus dur, désormais, vous pouvez attendre que d’autres traducteurs français regardent votre Pull Request, y apporter des suggestions, et l’accepte les modifications plus tard, ou vous pouvez aussi vous coller sur un autre article.

Reviewer: ```suggestion
À partir de ce moment, vous avez fait le plus dur. Vous pouvez désormais attendre que d’autres traducteurs français regardent votre Pull Request, y apportent des suggestions, et que les modifications soient approuvées, vous pouvez pendant ce temps commencer à traduire un autre article.
```

[b][size=150]Apporter des modifications à ma Pull Request[/size][/b]

> On vous à proposer des modifications à votre Pull Request ? voyez comment les accepter, ou y proposer des modifications si elle vous semble incorrecte.

Reviewer: ```suggestion
Des modifications ont été proposées sur votre Pull Request ? Voici comment les accepter, ou proposer des modifications si elles vous semblent incorrectes.
```

> Quand l’un des traducteurs (ou un maintainer), et pour faire avancer votre Pull Request dans le bon sens, vous devez sois appliquer les modifications que l’on vous a proposer, sois les commenter si vous n’êtes pas d’accord.

Reviewer: ```suggestion
Si l’un des traducteurs propose des modifications sur votre Pull Request, vous devez soit appliquer les modifications que l’on vous a proposées, soit les commenter si vous n’êtes pas d’accord.
```

Dans les deux cas, tout se passe dans l’onglet [b]Files changed[/b].
L’onglet ressemble à [b][url=https://i.imgur.com/YIUOdib.png]ceci[/url][/b].

> Pour faire un exemple, je vais simuler une demande de modification de ma propre Pull Request (c’est possible, et cela est plus simple que de repasser par Visual Studio Code pour apporter des modifications des suggestions – nous verrons cela un peu plus tard).

Reviewer: ```suggestion
Pour faire un exemple, je vais simuler une demande de modification de ma propre Pull Request (c’est possible, et cela est plus simple que de repasser par Visual Studio Code pour apporter des modifications – nous verrons cela un peu plus tard).
```

Reviewer: J'ai retiré "des suggestions" mais je suis pas sûr de ce que tu as voulu dire

Author: Ouais effectivement j'ai du me foirer, c'est mieux avec `des modifications` du coup x)

> Imaginons que je souhaite mettre une majuscule à [b]beatmapping[/b], qui est en ligne 7.

Reviewer: ```suggestion
Imaginons que je souhaite mettre une majuscule au mot [b]beatmapping[/b], qui se trouve à la ligne 7.
```

[b][url=https://i.imgur.com/IQ7JDku.png]Voici a quoi ressemblera la suggestion[/url][/b].

> Dans la partie rouge, c’est ce qui était auparavant avant la proposition (ce qui est souligner est l’élément que la personne qui proposer une modification agit, ici c’est [b]Beatmapping[/b].

Reviewer: ```suggestion
La partie rouge s'agit de ce qui se trouvait avant la proposition (ce qui est souligné est l’élément sur lequel la personne qui propose une modification agit, ici c’est [b]Beatmapping[/b].
```

> Dans la partie verte, c’est la version qui sera ajouter après que vous ayez accepter la modification.

Reviewer: ```suggestion
La partie verte est la version qui sera adoptée après acceptation des modifications.
```

> [b][url=https://i.imgur.com/eCuinvX.png]ici[/url][/b] vous voyez les modifications sur votre Pull Request, avec deux boutons : [b]Commit suggestion[/b] et [b]Add suggestion to batch[/b].

Reviewer: ```suggestion
Vous pouvez voir[b][url=https://i.imgur.com/eCuinvX.png]ici[/url][/b] les modifications sur votre Pull Request, avec deux boutons : [b]Commit suggestion[/b] et [b]Add suggestion to batch[/b].
```

> [b]Commit suggestion[/b] va appliquer directement la suggestion, après avoir donné un nom à votre commit. Pratique quand il n’y a qu’une seule suggestion de proposer, mais qui pose problème quand vous en avez plusieurs (comme ici)

Reviewer: ```suggestion
[b]Commit suggestion[/b] va appliquer directement la suggestion, après avoir donné un nom à votre commit. Pratique quand il n’y a qu’une seule suggestion à proposer, mais qui pose problème quand vous en avez plusieurs (comme ici)
```

> [b]Add suggestion to batch[/b] règlera votre problème si vous avez plusieurs suggestions, car il va vous prendre en compte que vous avez sélectionner cette modification, et quand vous en sélectionnez une autre, il la mettra aussi en attente.

Reviewer: ```suggestion
[b]Add suggestion to batch[/b] règlera votre problème si vous avez plusieurs suggestions, car il va prendre en compte que vous avez sélectionné cette modification, et quand vous en sélectionnez une autre, il la mettra aussi en attente.
```

> [b]Add suggestion to batch[/b] règlera votre problème si vous avez plusieurs suggestions, car il va vous prendre en compte que vous avez sélectionner cette modification, et quand vous en sélectionnez une autre, il la mettra aussi en attente.

Author: ```suggestion
[b]Add suggestion to batch[/b] règlera votre problème si vous avez plusieurs suggestions, car il va prendre en compte que vous avez sélectionné cette modification, et quand vous en sélectionnez une autre, il la mettra aussi en attente.

Ça évite donc de passer par Visual Studio Code, et est un gain de temps considérable dans le processus de review.
```

> Ensuite, quand vous aurez cliquez sur ce bouton, il sera remplacé par le précédent, avec un nombre à côté : il correspond au nombre de suggestions qui vont être appliquer et ajouter sur votre fichier. Suivez ensuite les étapes pour poster votre commit.

Reviewer: ```suggestion
Ensuite, quand vous aurez cliqué sur ce bouton, il sera remplacé par le précédent, avec un nombre à côté : il correspond au nombre de suggestions qui vont être appliquées sur votre fichier. Suivez ensuite les étapes pour poster votre commit.
```

> Cette solution est plus simple quand vous avez plusieurs suggestions, car cela évite d’inonder votre Pull Request de commit individuel, alors que vous pouvez appliquer toutes vos suggestions sur une seule et unique commit.

Reviewer: ```suggestion
Cette solution est plus simple quand vous avez plusieurs suggestions, car cela évite d’inonder votre Pull Request de commits individuelles, alors que vous pouvez appliquer toutes vos suggestions sur une seule et unique commit.
```

> [b][size=150]Proposer des modifications sur la Pull Request d’un contributeur[/size][/b]

Reviewer: ```suggestion
[b][size=150]Proposer des modifications sur la Pull Request d'un autre traducteur[/size][/b]
```

> Vous souhaitez proposer des modifications sur la Pull Request d’un autre contributeur, mais vous ne savez pas comment faire ? vous êtes au bon endroit ! Voyons ensemble comment procéder pour proposer une modification sur la Pull Request d’un contributeur.

Reviewer: ```suggestion
Vous souhaitez proposer des modifications sur la Pull Request d’un autre traducteur, mais vous ne savez pas comment faire. Voyons alors ensemble comment procéder pour proposer une modification sur la Pull Request d’un contributeur.
```

Author: J'aurais pas retirer le `?` du coup, ça fait bizarre

> La partie revue (review en anglais) est une partie très importante du processus de traduction. Il permet de déceler des fautes d’orthographe ou des problèmes qui ont été oubliés par le contributeur. Vous devez donc impérativement prendre cela au sérieux, cela nous évitera de créer des Pull Requests en pagaille pour corriger de petites fautes.

Reviewer: ```suggestion
La partie revue (review en anglais) est une partie très importante du processus de traduction. Elle permet de déceler des fautes d’orthographe ou d'autres problèmes qui ont été oubliés par le traducteur. Essayez donc de faire de votre mieux, cela nous évitera de créer des Pull Requests en pagaille pour corriger de petites fautes.
```

> Mais n’ayez crainte ! nous sommes humains, les erreurs arrivent, et si vous rater des fautes, ce n’est pas grave du tout !

Reviewer: ```suggestion
Mais n’ayez crainte ! Nous sommes humains, les erreurs arrivent, et si vous faites des erreurs, ce n’est pas grave du tout, les reviews sont là pour ça !
```

> Pour proposer des modifications sur la Pull Request d’un contributeur, commencez par ouvrir la page de sa Pull Request en cliquant dessus dans l’onglet [b]Pull requests[/b].

Reviewer: ```suggestion
Pour proposer des modifications sur la Pull Request d’un autre traducteur, commencez par ouvrir la page de sa Pull Request en cliquant dessus dans l’onglet [b]Pull requests[/b].
```

Dirigez-vous ensuite dans l’onglet [b]Files changed[/b].

Je vais rester sur la Pull Request que je viens de créer, qui servira d’exemple dans ce processus.

> Lisez attentivement l’article pour trouver de potentiels fautes.

Reviewer: ```suggestion
Lisez attentivement l’article pour trouver de potentielles fautes.
```

> Remarque : dans les articles qui modifie des fichiers déjà existants, concentrez-vous sur les parties en verte, et non celle en rouge, qui correspondent à ce qui a été enlevé. Si vous avez des difficultés, vous pouvez changer la vue de cet onglet en haut, en cliquant sur l’engrenage au milieu en haut, et en appuyant sur le bouton [b]Split[/b]. 

Reviewer: ```suggestion
Remarque : dans les Pull Requests qui modifient des fichiers déjà existants, concentrez-vous sur les parties en vert, et non celles en rouge, qui correspondent à ce qui a été enlevé. Si vous avez des difficultés, vous pouvez changer la vue de cet onglet en haut, en cliquant sur l’engrenage au milieu en haut, et en appuyant sur le bouton [b]Split[/b]. 
```

> Cette vue simplifiera la lecture des modifications pour ce type de fichiers, car les suppressions seront d’un côté, alors que les ajouts seront d’un autre côté. Ce qui ne sera pas modifier sera inchangé des deux côtés.

Reviewer: ```suggestion
Cette vue simplifiera la lecture des modifications pour ce type de fichiers, car les suppressions seront d’un côté, alors que les ajouts seront d’un autre côté. Ce qui ne sera pas modifié sera inchangé des deux côtés.
```

> Quand vous remarquez une faute, allez sur la ligne qui vous intéresse (le numéro des lignes est à gauche, dans la marge), vous verrez un petit [b]+[/b] qui va apparaître, cliquez dessus.

Reviewer: ```suggestion
Quand vous remarquez une faute, allez sur la ligne qui vous intéresse (le numéro des lignes est à gauche, dans la marge), vous verrez un petit [b]+[/b] apparaître, cliquez dessus.
```

Une fenêtre de dialogue va s’afficher, avec une zone de texte. Nous allons aller sur la barre d’outils en haut de cette zone de texte, et nous allons cliquer sur [b][url=https://i.imgur.com/BjXWtGj.png]ce bouton[/url][/b].

> Vous allez voir la ligne que vous avez sélectionner être entourer par une fonction [b]suggestion[/b], dans cette espace, modifiez ce qui doit être modifier. Quand vous aurez terminé, vous avez deux choix :

Reviewer: ```suggestion
La ligne que vous avez sélectionnée sera entourée par une fonction [b]suggestion[/b]. Dans cette espace, modifiez ce qui doit être modifié. Quand vous aurez terminé, vous avez deux choix :
```

[list]

> [*]Un bouton [b]Add single comment[/b] qui va directement vous poster la suggestion

Reviewer: ```suggestion
[*]Un bouton [b]Add single comment[/b] qui va directement poster la suggestion
```

> [*]Un bouton [b]Start a review[/b] qui va vous mettre votre suggestion en attente, et vous permettre de proposer d’autres modifications si vous en avez découvert.

Reviewer: ```suggestion
[*]Un bouton [b]Start a review[/b] qui va mettre votre suggestion en attente, et vous permettre de proposer d’autres modifications si vous en avez découvert.
```

[/list]

> Si vous avez terminé votre review avec la fonction [b]Start a review[/b], retourner en haut de la page, et cliquez sur [b][url=https://i.imgur.com/BafVZ2D.png]ce bouton[/url][/b], puis, sélectionner la case [b]Request changes[/b] pour indiquer que votre commentaire porte sur la suggestion de modifications sur la Pull Request.

Reviewer: ```suggestion
Si vous avez terminé votre review avec la fonction [b]Start a review[/b], retournez en haut de la page, et cliquez sur [b][url=https://i.imgur.com/BafVZ2D.png]ce bouton[/url][/b]. Sélectionnez alors la case [b]Request changes[/b] pour indiquer que votre commentaire porte sur la suggestion de modifications sur la Pull Request.
```

> Si vous estimez que la Pull Request est correcte, au lieu de sélectionner [b]Request changes[/b], cliquez sur [b]Approve[/b] pour approuver les modifications proposées par ce contributeur.

Reviewer: ```suggestion
Si vous estimez que la Pull Request est correcte, au lieu de sélectionner [b]Request changes[/b], cliquez sur [b]Approve[/b] pour approuver les modifications proposées par ce traducteur.
```

> Nous avons fait le tour de tout ce que vous devez savoir sur le processus de traduction. Cela a été un peu long, mais il est important de détailler tout le processus dans son ensemble, afin que vous puissiez le mieux comprendre ce processus, qui reste complexe sans des explications claires et précises.

Reviewer: ```suggestion
Nous avons fait le tour de tout ce que vous devez savoir sur le processus de traduction. Cela a été un peu long, mais il est important de détailler tout le processus dans son ensemble, afin que vous puissiez comprendre autant que possible ce processus qui reste complexe sans des explications claires et précises.
```

> Ce tutoriel a été écrit et imaginer par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], compléter et améliorer par ....

Reviewer: ```suggestion
Ce tutoriel a été écrit et imaginé par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], complété et amélioré par ....
```

> Ce tutoriel a été écrit et imaginer par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], compléter et améliorer par ....

Author: ```suggestion
Ce tutoriel a été écrit et imaginé par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], complété et amélioré par [b][url=https://osu.ppy.sh/users/10660738]Ganondorf[/url][/b]
```

[b][size=150]FAQ[/size][/b]

> [b]Question :[/b] J'ai créé une Pull Request de mes modifications, un contributeur français à approuvés les modifications que j'ai faite, quel est la suite ?

Reviewer: ```suggestion
[b]Question :[/b] J'ai créé une Pull Request de mes modifications, un contributeur français à approuvé les modifications que j'ai faites, et maintenant ?
```

> [b]Réponse :[/b] Si un contributeur français a approuvé vos modifications sur votre Pull Request, vous pouvez sois attendre qu'un Maintainers fusionne votre Pull Request (on appelle ça aussi [b]Merge[/b]), sois vous pouvez demander à ces personnes de le faire en leur envoyant un message privé sur leur Discord (voir "liens utile" en fin de post).

Reviewer: ```suggestion
[b]Réponse :[/b] Si un contributeur français a approuvé vos modifications sur votre Pull Request, vous pouvez soit attendre qu'un maintainer fusionne votre Pull Request (on appelle ça aussi [b]Merge[/b]), soit demander à ces personnes de le faire en leur envoyant un message privé sur leur Discord (voir "liens utiles" en fin de post).
```

[b]Question :[/b] On me dit que ma branche est [i]"obsolète"[/i], qu'est-ce que cela signifie ?

> [b]Réponse :[/b] Une branche obsolète est [i]"normal"[/i], car quand vous avez cloner le projet sur une version que vous pouvez modifier, vous l'avait faite à un instant T, et cette dernière ne se mettant pas à jour automatiquement, vous devez le faire manuellement.

Reviewer: ```suggestion
[b]Réponse :[/b] Une branche obsolète est [i]"normale"[/i], car lorsque vous avez cloné le projet sur une version que vous pouvez modifier, vous l'avait fait à un instant T, et cette dernière ne se mettant pas à jour automatiquement, vous devez le faire manuellement.
```

Pour ce faire : 

[list]

> [*]Allez sur votre fork (vous pouvez y accéder en allant sur la page d'accueil du dépôt, et cliquer sur [b]Fork[/b] en haut à droite de la page)

Reviewer: ```suggestion
[*]Allez sur votre fork (vous pouvez y accéder en allant sur la page d'accueil du dépôt, puis en cliquant sur [b]Fork[/b] en haut à droite de la page)
```

[*]Cliquez sur [b]branches[/b] en haut à gauche
[*]Cherchez la branche qui pose problème

> [*]Quand vous serez sur votre branche, regardez-en haut de la page, en dessous de [b]branches[/b] le message qu'il est écrit.

Reviewer: ```suggestion
[*]Quand vous serez sur votre branche, regardez en haut de la page, en dessous de [b]branches[/b], un message sera écrit.
```

> Si votre branche à [b][url=https://i.imgur.com/8wIrVwP.jpg]ce statuts[/url][/b], vous n'avez rien à faire : elle est à jour

Reviewer: ```suggestion
Si votre branche a [b][url=https://i.imgur.com/8wIrVwP.jpg]ce statut[/url][/b], vous n'avez rien à faire : elle est à jour
```

> Si votre branche à [b][url=https://i.imgur.com/u9wICMw.jpg]ce statuts[/url][/b], vous n'avez rien à faire : ce sont les modifications que vous avez apportées

Reviewer: ```suggestion
Si votre branche a [b][url=https://i.imgur.com/u9wICMw.jpg]ce statut[/url][/b], vous n'avez rien à faire : ce sont les modifications que vous avez apportées
```

> Si votre branche à [b][url=https://i.imgur.com/GVoiq0L.jpg]ce statuts[/url][/b], votre branche a du retard sur la branche d'origine, vous devez donc la mettre à jour.

Reviewer: ```suggestion
Si votre branche a [b][url=https://i.imgur.com/GVoiq0L.jpg]ce statut[/url][/b], votre branche a du retard sur la branche d'origine, vous devez donc la mettre à jour.
```

> Si votre branche à [b][url=https://i.imgur.com/oVvkgDM.jpg]ce statuts[/url][/b], votre branche a du retard et de l'avance sur la branche d'origine, vous devez la mettre à jour car il y a du retard.

Reviewer: ```suggestion
Si votre branche a [b][url=https://i.imgur.com/oVvkgDM.jpg]ce statut[/url][/b], votre branche a du retard et de l'avance sur la branche d'origine, vous devez la mettre à jour car il y a du retard.
```

> [*]Si votre statuts correspond à l'un des deux qui signifie que votre branche n'est pas à jour, cliquez sur [b][url=https://i.imgur.com/F3N0zrg.png]ce bouton[/url][/b] qui est à gauche du message, et ensuite cliquer sur [b]Fetch and merge[/b] pour mettre à jour votre branche.

Reviewer: ```suggestion
[*]Si votre statut correspond à l'un des deux qui signifient que votre branche n'est pas à jour, cliquez sur [b][url=https://i.imgur.com/F3N0zrg.png]ce bouton[/url][/b] qui est à gauche du message, cliquez ensuite sur [b]Fetch and merge[/b] pour mettre à jour votre branche.
```

> [*]Pensez également à mettre à jour de temps en temps votre branche [b]master[/b], c'est sur cette branche que va se baser toutes les autres, vous devez donc essayez de la garder la plus à jour possible.

Reviewer: ```suggestion
[*]Pensez également à mettre à jour de temps en temps votre branche [b]master[/b], c'est sur cette branche que va se baser toutes les autres, vous devez donc essayer de la garder le plus à jour possible.
```

[/list]
[b]Question :[/b] Après avoir mis à jour ma branche, j'ai des conflits, mais je ne sais pas comment les résoudre !

> [b]Réponse :[/b] Les conflits correspondent à des modifications qui ont été merge avant les vôtre sur les fichiers que vous modifiez, et ce sont ces modifications qui entrent en conflit avec les vôtres.

Reviewer: ```suggestion
[b]Réponse :[/b] Les conflits correspondent à des modifications qui ont été merged avant les vôtres sur les fichiers que vous modifiez, et ce sont ces modifications qui entrent en conflit avec les vôtres.
```

Pour les résoudre : (réalisé à l'aide du guide du contributeur sur l'osu! wiki, nous avons besoin d'un cas concret d'un contributeur pour réaliser une procédure plus explicite)
[list]
[*] Cliquez sur le bouton [b]Resolve conflicts[/b]

> [*] Vous verrez des bouts de texte qui sont entourer avec [b]<<<<<<<[/b] et [b]=======[/b] : ce sont vos modifications

Reviewer: ```suggestion
[*] Vous verrez des bouts de texte qui sont entourés avec [b]<<<<<<<[/b] et [b]=======[/b] : ce sont vos modifications
```

> [*] Vous verrez des bouts de texte qui sont entourer avec [b]=======[/b] et [b]>>>>>>> master[/b] : ce sont les modifications qui sont en conflit avec les vôtres

Reviewer: ```suggestion
[*] Vous verrez des bouts de texte qui sont entourés avec [b]=======[/b] et [b]>>>>>>> master[/b] : ce sont les modifications qui sont en conflit avec les vôtres
```

> [*] Pour résoudre le conflit, retirer tous les marqueurs présents en surbrillance, en prenant garde que les modifications [i]"entrantes"[/i] ne soit pas en double avec celle déjà présente ([i]"sortantes"[/i] donc), sinon, retirer la partie que vous aviez réalisé avant votre conflit (si les modifications en conflit sont le but de votre Pull Request, conservez-les)

Reviewer: ```suggestion
[*] Pour résoudre le conflit, retirez tous les marqueurs présents en surbrillance, en prenant garde que les modifications [i]"entrantes"[/i] ne soient pas en double avec celles déjà présentes ([i]"sortantes"[/i] donc), sinon, retirez la partie que vous aviez réalisé avant votre conflit (si les modifications en conflit sont le but de votre Pull Request, conservez-les)
```

[*] Si tous les conflits sont résolus, vous pouvez de nouveau cliquer sur le bouton [b]Resolve conflicts[/b].

> Si ce bouton est grisé, vous devez mettre à jour votre branche (le cas échéant) avant de résoudre le conflit

Reviewer: ```suggestion
Si ce bouton est grisé, vous devez mettre à jour votre branche (le cas échéant) avant de résoudre le conflit.
```

[/list]

> À noter : ce cas est courant si l'on modifie des fichiers [b]en.md[/b], les conflits sur les traductions sont rares, sauf si la personne n'avait pas vu qu'il y avait déjà une Pull Request, mais normalement nous vérifions que cela n'arrive pas, et corrigions le problème le plus vite possible afin d'éviter ce cas de figure.

Reviewer: ```suggestion
À noter : ce cas est courant si l'on modifie des fichiers [b]en.md[/b], les conflits sur les traductions sont rares, sauf si la personne n'avait pas vu qu'il y avait déjà une Pull Request, mais normalement nous vérifions que cela n'arrive pas, et corrigeons le problème le plus vite possible afin d'éviter ce cas de figure.
```

> [b]Question :[/b] J'aimerais changer de branche, mais j'ai déjà des modifications en cours dessus, comment je peux faire ?

Reviewer: ```suggestion
[b]Question :[/b] J'aimerais changer de branche, mais j'ai déjà des modifications en cours dessus, comment puis-je faire ?
```

[b]Réponse :[/b] Pour une raison ou une autre, vous souhaitez changer de branche, ou bien en créer une autre pour faire autre chose. Cependant, vous avez des modifications sur celle actuellement active.

> GitHub Desktop vous informera donc via [b][url=https://i.imgur.com/gouL9ie.png]cette fenêtre[/url][/b] si vous souhaitez laisser vos modifications sur cette branche ([b]Leave my changes on {nom de votre branche}[/b]), ce qui mettra les modifications sur cette branche en attente, et récupérable par la suite, ou si vous souhaitez que vos modifications sur la branche actuelle soit suivies sur la branche que vous avez sélectionnée ([b]Bring my changes to {nom de votre branche}[/b]).

Reviewer: ```suggestion
GitHub Desktop vous informera donc via [b][url=https://i.imgur.com/gouL9ie.png]cette fenêtre[/url][/b] si vous souhaitez laisser vos modifications sur cette branche ([b]Leave my changes on {nom de votre branche}[/b]), ce qui mettra les modifications sur cette branche en attente, et récupérables par la suite, ou si vous souhaitez que vos modifications sur la branche actuelle soient suivies sur la branche que vous avez sélectionnée ([b]Bring my changes to {nom de votre branche}[/b]).
```

Selon ce que vous souhaitez faire, sélectionner l'une des deux options.

> Quand vous reviendrez sur votre branche en ayant sélectionné [b]Leave my changes on {nom de votre branche}[/b], vous avez un [b][url=https://i.imgur.com/QSYgQfe.png]bouton[/url][/b] qui se trouve dans la zone réservée habituellement a l'affichage des changements effectuer.

Reviewer: ```suggestion
Quand vous reviendrez sur votre branche en ayant sélectionné [b]Leave my changes on {nom de votre branche}[/b], vous avez un [b][url=https://i.imgur.com/QSYgQfe.png]bouton[/url][/b] qui se trouve dans la zone réservée habituellement a l'affichage des changements effectués.
```

> Cliquez sur ce bouton. Vous allez arriver [b][url=https://i.imgur.com/HL0h0ZM.png]ici[/url][/b]. Vous avez deux choix : [b]Restore[/b] pour restaurer les modifications qui s'affiche, [b]Discard[/b] pour abandonner les modifications qui s'affiche, elles seront totalement perdues après cela.

Reviewer: ```suggestion
Cliquez sur ce bouton. Vous allez arriver [b][url=https://i.imgur.com/HL0h0ZM.png]ici[/url][/b]. Vous avez deux choix : [b]Restore[/b] pour restaurer les modifications qui s'affiche, [b]Discard[/b] pour abandonner les modifications affichées, elles seront totalement perdues après cela.
```

[b][size=150]Liens utiles[/size][/b]

[b][url=https://discord.com/invite/vfrHNQZ]Lien vers le Discord de la communauté française d'osu![/url][/b] (un salon #traduction est réservé à la traduction). Vous avez besoin d'un compte osu! afin de rejoindre le Discord, une vérification sera faite par les modérateurs.

> [b][url=https://github.com/ppy/osu-wiki]Dépôt du projet osu-wiki[/url][/b]

Author: ```suggestion
[b][url=https://github.com/ppy/osu-wiki]Dépôt du projet osu-wiki[/url][/b]
[b][url=https://docs.google.com/spreadsheets/d/1zppXOH3PK32c0FwR8TC8QixxzoliEPcWNOjMfAfAv8o/edit#gid=1581693581]Spreadsheet du projet[/url]
```

[b]Commit suggestion[/b] va appliquer directement la suggestion, après avoir donné un nom à votre commit. Pratique quand il n’y a qu’une seule suggestion à proposer, mais qui pose problème quand vous en avez plusieurs (comme ici)

[b]Add suggestion to batch[/b] règlera votre problème si vous avez plusieurs suggestions, car il va prendre en compte que vous avez sélectionné cette modification, et quand vous en sélectionnez une autre, il la mettra aussi en attente.

Ça évite donc de passer par Visual Studio Code, et est un gain de temps considérable dans le processus de review.

Ensuite, quand vous aurez cliqué sur ce bouton, il sera remplacé par le précédent, avec un nombre à côté : il correspond au nombre de suggestions qui vont être appliquées sur votre fichier. Suivez ensuite les étapes pour poster votre commit.
Cette solution est plus simple quand vous avez plusieurs suggestions, car cela évite d’inonder votre Pull Request de commits individuelles, alors que vous pouvez appliquer toutes vos suggestions sur une seule et unique commit.

[b][size=150]Proposer des modifications sur la Pull Request d'un autre traducteur[/size][/b]

> Vous souhaitez proposer des modifications sur la Pull Request d’un autre contributeur, mais vous ne savez pas comment faire ? vous êtes au bon endroit ! Voyons ensemble comment procéder pour proposer une modification sur la Pull Request d’un contributeur.

Author: Vous souhaitez proposer des modifications sur la Pull Request d’un autre traducteur, mais vous ne savez pas comment faire ? Voyons alors ensemble comment procéder pour proposer une modification sur la Pull Request d’un contributeur.

GitHub Desktop vous informera donc via [b][url=https://i.imgur.com/gouL9ie.png]cette fenêtre[/url][/b] si vous souhaitez laisser vos modifications sur cette branche ([b]Leave my changes on {nom de votre branche}[/b]), ce qui mettra les modifications sur cette branche en attente, et récupérables par la suite, ou si vous souhaitez que vos modifications sur la branche actuelle soient suivies sur la branche que vous avez sélectionnée ([b]Bring my changes to {nom de votre branche}[/b]).
Selon ce que vous souhaitez faire, sélectionner l'une des deux options.

Quand vous reviendrez sur votre branche en ayant sélectionné [b]Leave my changes on {nom de votre branche}[/b], vous avez un [b][url=https://i.imgur.com/QSYgQfe.png]bouton[/url][/b] qui se trouve dans la zone réservée habituellement a l'affichage des changements effectués.
Cliquez sur ce bouton. Vous allez arriver [b][url=https://i.imgur.com/HL0h0ZM.png]ici[/url][/b]. Vous avez deux choix : [b]Restore[/b] pour restaurer les modifications qui s'affiche, [b]Discard[/b] pour abandonner les modifications affichées, elles seront totalement perdues après cela.

[b][size=150]Liens utiles[/size][/b]

> [b][url=https://discord.com/invite/vfrHNQZ]Lien vers le Discord de la communauté française d'osu![/url][/b] (un salon #traduction est réservé à la traduction). Vous avez besoin d'un compte osu! afin de rejoindre le Discord, une vérification sera faite par les modérateurs.
> [b][url=https://github.com/ppy/osu-wiki]Dépôt du projet osu-wiki[/url][/b]
> [b][url=https://docs.google.com/spreadsheets/d/1zppXOH3PK32c0FwR8TC8QixxzoliEPcWNOjMfAfAv8o/edit#gid=1581693581]Spreadsheet du projet[/url][/b]

Author: ```suggestion
[list]
[*][b][url=https://discord.com/invite/vfrHNQZ]Lien vers le Discord de la communauté française d'osu![/url][/b] (un salon #traduction est réservé à la traduction). Vous avez besoin d'un compte osu! afin de rejoindre le Discord, une vérification sera faite par les modérateurs.
[*][b][url=https://github.com/ppy/osu-wiki]Dépôt du projet osu-wiki[/url][/b]
[*][b][url=https://docs.google.com/spreadsheets/d/1zppXOH3PK32c0FwR8TC8QixxzoliEPcWNOjMfAfAv8o/edit#gid=1581693581]Spreadsheet du projet[/url][/b]
```

Enfin, il va vous proposer [b][url=https://imgur.com/gVZPZyn.png]de cloner le dépôt sur votre ordinateur[/url][/b]. Si tout va bien, vous aurez à droite la liste des dépôts qui sont présents sur votre compte. Cliquez sur [b]{votre nom d’utilisateur}/osu-wiki[/b], puis sur [b]Clone{votre nom d’utilisateur}/osu-wiki[/b], et une nouvelle fois sur [b]Clone[/b].

Si rien ne s'affiche, cliquez sur le bouton [b]Clone a repository from the Internet…[/b], [b][url=https://imgur.com/GVtkqIZ.png]une petite fenêtre va s’ouvrir[/url][/b], si vous ne voyez toujours rien, allez dans la section [b]URL[/b], et tapez [b]{votre nom d’utilisateur}/osu-wiki[/b].
Si tout va bien, le projet devrait se télécharger sur votre ordinateur. Cela sera également long, car le projet est volumineux.

Lorsque le projet aura été copié sur votre ordinateur, vous devriez avoir une fenêtre qui ressemble à [b][url=https://imgur.com/Sz5CYFB.png]celle-ci[/url][/b]. Cette fenêtre va vous demander comment vous comptez utiliser votre fork.
[list]

> [*][b]To contribute to the parent project[/b] pour contribuer sur le projet « parent », il s'agira dans notre cas de [b]ppy/osu-wiki[/b]

Author: ```suggestion
[*][b]To contribute to the parent project[/b] pour contribuer sur le projet [i]parent[/i] il s'agira dans notre cas de [b]ppy/osu-wiki[/b]
```

[*]Quand vous devez créer un fichier pour accueillir la traduction française d’une page, appelez votre fichier [b]fr.md[/b].

[*]Un conseil : quand vous souhaitez traduire une page, je vous conseille de prendre tout le contenu du fichier original en anglais, de le copier sur votre fichier de traduction, et de traduire directement dessus. Ça vous évite de devoir refaire toute la mise en forme, et c’est plus rapide.
[/list]

Maintenant que vous savez tout ça, vous pouvez vous lancer.

[b][size=150]Mettre en ligne la contribution[/size][/b]

Une fois votre traduction terminée, revenez sur GitHub Desktop, la fenêtre principale [b][url=https://imgur.com/QqLYndG.png]aura légèrement changé[/url][/b].
Sur la gauche, vous avez la liste des fichiers que vous avez créés, supprimés et modifiés. Un [b]+[/b] vert apparaîtra à côté du nom du fichier s'il s'agit d'un fichier créé, un [b].[/b] orange s'il s'agit d'un fichier modifié, et un [b]-[/b] rouge s'il s'agit d'un fichier supprimé.

> Au milieu, vous avez le contenu du fichier, avec en vert tout ce que vous aurez ajouté, et en rouge tout ce qui aura été enlevé.
> Avant de la mettre en ligne, vous devez créer un [b]commit[/b] de votre modification.
> Un commit est un terme qu’on utilise pour désigner une modification que l’on apporte à une branche, par rapport à la version qui est déjà sur le projet.

Author: ```suggestion
Au milieu, vous avez le contenu du fichier, avec en vert tout ce que vous aurez ajouté, et en rouge tout ce qui aura été enlevé.

Avant de la mettre en ligne, vous devez créer un [b]commit[/b] de votre modification.

Un commit est un terme qu’on utilise pour désigner une modification que l’on apporte à une branche, par rapport à la version qui est déjà sur le projet.
```

Pour créer ce commit, vous devez vous diriger en bas à gauche de la fenêtre de GitHub Desktop, plus particulièrement [b][url=https://imgur.com/fVRnAwB.png]ici[/url][/b].

> Dans la barre à côté de votre avatar, vous devez spécifier un titre à votre commit. Nous vous conseillons de mettre ce genre de titre : [b][FR]{Add}/{Update} {`Nom de l’article`}[/b].

Author: ```suggestion
Dans la zone de texte à côté de votre photo de profil, vous devez spécifier un titre à votre commit. Nous vous conseillons de mettre ce genre de titre : [b][FR]{Add}/{Update} {`Nom de l’article`}[/b].
```

Dans mon cas, je vais mettre le titre suivant : [b][FR] Add `Antiflow`[/b], car j’ai ajouté une page, et j’ai ajouté la traduction de l’article [b]Antiflow[/b].

> Dans la zone de texte plus bas, vous pouvez ajouter si vous le souhaitez une description de ce que vous avez fait (en anglais), mais cela n’est pas obligatoire.
> Quand tout est bon, vous pouvez cliquer sur [b]Commit to {nom de votre branche}[/b]. Vous verrez apparaître [url=https://imgur.com/nXB4upz.png]ceci[/url]. Cela veut dire que votre commit a bien fonctionné, et qu’elle est sur votre branche, elle est alors en attente.

Author: ```suggestion
Dans la zone de texte en dessous, vous pouvez ajouter si vous le souhaitez une description de ce que vous avez fait (en anglais), mais cela n’est pas obligatoire.

Quand tout est bon, vous pouvez cliquer sur [b]Commit to {nom de votre branche}[/b]. Vous verrez apparaître [url=https://imgur.com/nXB4upz.png]ceci[/url]. Cela veut dire que votre commit a bien fonctionné, et qu’elle est sur votre branche, elle est alors en attente.
```

> Puisqu'il s'agit de votre premier commit, vous n’avez pas de [b]Pull Request[/b] active, vous devez donc en créer une.

Author: ```suggestion
Puisqu'il s'agit de votre premier commit, vous n’avez pas de [i]Pull Request[/i] active, vous devez donc en créer une.
```

> Une Pull Request (PR en abrégé), c’est l’endroit où vous allez proposer votre modification sur le projet, pour qu’elle soit critiquée et qu’on vous propose des modifications. Le cas échéant, vos modifications seront approuvées si nous n’avons rien à redire.

Author: ```suggestion
Une Pull Request (abrégé en PR), c’est l’endroit où vous allez proposer votre modification sur le projet, pour qu’elle soit critiquée et qu’on vous propose des modifications. Le cas échéant, vos modifications seront approuvées si nous n’avons rien à redire.
```

> Pour en créer une, nous resterons sur GitHub Desktop, et remarquer que la fenêtre d’origine a [b][url=https://imgur.com/JoQKxOC.png]légèrement changé[/url][/b].
> Comme vous pouvez le voir, un message bleu vous dira ce qui a été dit plus haut : que votre branche n’est pas publiée sur GitHub, et n’a donc pas de Pull Request à son actif. Pour la créer, vous pouvez cliquer sur [b]Publish Branch[/b] à droite du message, ou cliquer en haut sur [b]Publish Branch[/b], juste à côté de l'endroit où vous avez créé votre branche. Les deux actions amènent au même résultat.

Author: ```suggestion
Pour en créer une, nous resterons sur GitHub Desktop, et remarquer que la fenêtre d’origine a [b][url=https://imgur.com/JoQKxOC.png]légèrement changé[/url][/b].

Comme vous pouvez le voir, un message bleu vous dira ce qui a été dit plus haut : que votre branche n’est pas publiée sur GitHub, et n’a donc pas de Pull Request à son actif. Pour la créer, vous pouvez cliquer sur [b]Publish Branch[/b] à droite du message, ou cliquer en haut sur [b]Publish Branch[/b], juste à côté de l'endroit où vous avez créé votre branche. Les deux actions amènent au même résultat.
```

> Une fois que vous aurez cliqué sur l’un des deux boutons, votre branche sera être publiée (celle s'appelle un [b]push[/b]) sur GitHub.

Author: ```suggestion
Une fois que vous aurez cliqué sur l’un des deux boutons, votre branche sera être publiée (cela s'appelle un [b]push[/b]) sur GitHub.
```

> Désormais, vous pouvez revenir sur [b][url=https://github.com/ppy/osu-wiki]l’accueil du dépôt[/url][/b], ou rester sur GitHub Desktop et cliquer sur [b]Create Pull Request[/b] au centre de la fenêtre, au même endroit où il vous propose de publier la branche sur GitHub.

Author: ```suggestion
Désormais, vous pouvez revenir sur [b][url=https://github.com/ppy/osu-wiki]l’accueil du dépôt[/url][/b], ou rester sur GitHub Desktop et cliquer sur [b]Create Pull Request[/b] au centre de la fenêtre, au même endroit où il vous proposais de publier la branche sur GitHub.
```

Si vous choisissez de retourner sur le dépôt, vous pourrez observer un [b][url=https://imgur.com/1sSHcQZ.png]petit changement[/url][/b] en haut de votre écran. Cette fenêtre en orange est la branche que vous venez juste de publier sur GitHub, cliquez donc sur [b]Compare & pull request[/b].

Vous allez arriver sur [b][url=https://imgur.com/9ENAr8z.png]cette fenêtre[/url][/b]. C’est là que vous pouvez voir les informations sur votre branche, avec les modifications que vous voulez proposer.

> Ce qui se trouve tout en haut, ne vous en occupez pas, ce sont simplement les paramètres qui définissent sur quel dépôt du projet il doit se baser pour qu’il soit publié par la suite.

Author: ```suggestion
Ce qui se trouve tout en haut, ne vous en occupez pas, ce sont simplement les paramètres qui définissent sur quel dépôt du projet il doit se baser pour déterminer la différence entre les deux versions du fichier.
```

Juste en dessous, vous avez le titre de votre premier commit, qui est automatiquement transformé en votre titre pour votre Pull Request, avec la description en dessous (si vous en avez ajouté une).
Si vous allez en dessous, vous avez le même affichage que tout à l’heure sur GitHub Desktop, avec ce que vous avez ajouté au fichier (ou supprimé).

> Une petite information importante : en bas de la zone de texte pour la description, vous avez [b][url=https://imgur.com/1ucKWCA.png]ceci[/url][/b]. Ce paramètre doit être coché, car il permet aux modérateurs du projet (ils sont appelés [b]Maintainers[/b]) d’accéder à votre branche et d’y apporter des modifications, comme ce que vous ferez plus tard. Cela n’est pas gênant de leur laisser l’accès à votre branche, car seuls eux pourront y accéder.
> Lorsque vous aurez ajouté une description (vous pouvez simplement dire que vous avez traduit tel article, cela suffira), cliquez sur [b]Create pull request[/b].

Author: ```suggestion
Une petite information importante : en bas de la zone de texte pour la description, vous avez [b][url=https://imgur.com/1ucKWCA.png]ceci[/url][/b]. Ce paramètre doit être coché, car il permet aux modérateurs du projet (ils sont appelés [b]Maintainers[/b]) d’accéder à votre branche et d’y apporter des modifications, comme ce que vous ferez plus tard. Cela n’est pas gênant de leur laisser l’accès à votre branche, car seuls eux pourront y accéder.

Lorsque vous aurez ajouté une description (vous pouvez simplement dire que vous avez traduit tel article, cela suffira), cliquez sur [b]Create pull request[/b].
```

Une fois votre Pull Request publiée, voici [b][url=https://imgur.com/VsIAK7L.png]à quoi cela ressemble[/url][/b].
À partir de ce moment, vous avez fait le plus dur. Vous pouvez désormais attendre que d’autres traducteurs français regardent votre Pull Request, y apportent des suggestions, et que les modifications soient approuvées, vous pouvez pendant ce temps commencer à traduire un autre article.

[b][size=150]Apporter des modifications à ma Pull Request[/size][/b]

Des modifications ont été proposées sur votre Pull Request ? Voici comment les accepter, ou proposer des modifications si elles vous semblent incorrectes.

> Si l’un des traducteurs propose des modifications sur votre Pull Request, vous devez soit appliquer les modifications que l’on vous a proposées, soit les commenter si vous n’êtes pas d’accord.
> Dans les deux cas, tout se passe dans l’onglet [b]Files changed[/b].
> L’onglet ressemble à [b][url=https://imgur.com/YIUOdib.png]ceci[/url][/b].

Author: ```suggestion
Si l’un des traducteurs propose des modifications sur votre Pull Request, vous devez soit appliquer les modifications que l’on vous a proposées, soit les commenter si vous n’êtes pas d’accord.
Dans les deux cas, tout se passe dans l’onglet [b]Files changed[/b].

L’onglet ressemble à [b][url=https://imgur.com/YIUOdib.png]ceci[/url][/b].
```

> Pour faire un exemple, je vais simuler une demande de modification de ma propre Pull Request (c’est possible, et cela est plus simple que de repasser par Visual Studio Code pour apporter des modifications – nous verrons cela un peu plus tard).

Author: ```suggestion
Pour faire un exemple, je vais simuler une demande de modification de ma propre Pull Request (c’est possible, et cela est plus simple que de repasser par Visual Studio Code pour apporter des modifications ; nous verrons cela un peu plus tard).
```

Imaginons que je souhaite mettre une majuscule au mot [b]beatmapping[/b], qui se trouve à la ligne 7.
[b][url=https://imgur.com/IQ7JDku.png]Voici a quoi ressemblera la suggestion[/url][/b].

La partie rouge s'agit de ce qui se trouvait avant la proposition (ce qui est souligné est l’élément sur lequel la personne qui propose une modification agit, ici c’est [b]Beatmapping[/b].

La partie verte est la version qui sera adoptée après acceptation des modifications.

> Vous pouvez voir[b][url=https://imgur.com/eCuinvX.png]ici[/url][/b] les modifications sur votre Pull Request, avec deux boutons : [b]Commit suggestion[/b] et [b]Add suggestion to batch[/b].

Author: ```suggestion
Vous pouvez voir [b][url=https://imgur.com/eCuinvX.png]ici[/url][/b] les modifications sur votre Pull Request, avec deux boutons : [b]Commit suggestion[/b] et [b]Add suggestion to batch[/b].
```

> 
> [b]Commit suggestion[/b] va appliquer directement la suggestion, après avoir donné un nom à votre commit. Pratique quand il n’y a qu’une seule suggestion à proposer, mais qui pose problème quand vous en avez plusieurs (comme ici)
> 
> [b]Add suggestion to batch[/b] règlera votre problème si vous avez plusieurs suggestions, car il va prendre en compte que vous avez sélectionné cette modification, et quand vous en sélectionnez une autre, il la mettra aussi en attente.
> 

Author: ```suggestion
[list]
[*][b]Commit suggestion[/b] va appliquer directement la suggestion, après avoir donné un nom à votre commit. Pratique quand il n’y a qu’une seule suggestion à proposer, mais qui pose problème quand vous en avez plusieurs (comme ici)
[*][b]Add suggestion to batch[/b] règlera votre problème si vous avez plusieurs suggestions, car il va prendre en compte que vous avez sélectionné cette modification, et quand vous en sélectionnez une autre, il la mettra aussi en attente.
Au final, vous aurez vos deux suggestions en un seul commit, plutôt qu'en deux séparé. L'avantage, c'est que cela rend votre Pull Request propre ; l'inconvénient : y'en a pas !
[/list]
```

> Ça évite donc de passer par Visual Studio Code, et est un gain de temps considérable dans le processus de review.
> 
> Ensuite, quand vous aurez cliqué sur ce bouton, il sera remplacé par le précédent, avec un nombre à côté : il correspond au nombre de suggestions qui vont être appliquées sur votre fichier. Suivez ensuite les étapes pour poster votre commit.
> Cette solution est plus simple quand vous avez plusieurs suggestions, car cela évite d’inonder votre Pull Request de commits individuelles, alors que vous pouvez appliquer toutes vos suggestions sur une seule et unique commit.
> 

Author: ```suggestion
Ensuite, quand vous aurez cliqué sur ce bouton, il sera remplacé par le précédent, avec un nombre à côté : il correspond au nombre de suggestions qui vont être appliquées sur votre fichier. Suivez ensuite les étapes pour poster votre commit.

Ces solutions évitent de repasser par Visual Studio Code, de chercher la ligne ou on vous propose une modification, appliquer sa modification, sauvegarder, faire un commit.
Cela prend un temps considérable. Avec ces solutions, vous gagnez un temps considérable dans le processus de review, qui peut être long selon le volume du fichier.

```

> [b][size=150]Proposer des modifications sur la Pull Request d'un autre traducteur[/size][/b]

Author: ```suggestion
[b][size=150]Proposer des modifications sur la Pull Request d'un autre contributeur[/size][/b]
```

Vous souhaitez proposer des modifications sur la Pull Request d’un autre traducteur, mais vous ne savez pas comment faire ? Voyons alors ensemble comment procéder pour proposer une modification sur la Pull Request d’un contributeur.

La partie revue (review en anglais) est une partie très importante du processus de traduction. Elle permet de déceler des fautes d’orthographe ou d'autres problèmes qui ont été oubliés par le traducteur. Essayez donc de faire de votre mieux, cela nous évitera de créer des Pull Requests en pagaille pour corriger de petites fautes.

Mais n’ayez crainte ! Nous sommes humains, les erreurs arrivent, et si vous faites des erreurs, ce n’est pas grave du tout, les reviews sont là pour ça !

Pour proposer des modifications sur la Pull Request d’un autre traducteur, commencez par ouvrir la page de sa Pull Request en cliquant dessus dans l’onglet [b]Pull requests[/b].
Dirigez-vous ensuite dans l’onglet [b]Files changed[/b].

> Je vais rester sur la Pull Request que je viens de créer, qui servira d’exemple dans ce processus.
> Lisez attentivement l’article pour trouver de potentielles fautes.
> Remarque : dans les Pull Requests qui modifient des fichiers déjà existants, concentrez-vous sur les parties en vert, et non celles en rouge, qui correspondent à ce qui a été enlevé. Si vous avez des difficultés, vous pouvez changer la vue de cet onglet en haut, en cliquant sur l’engrenage au milieu en haut, et en appuyant sur le bouton [b]Split[/b]. 
> 
> Cette vue simplifiera la lecture des modifications pour ce type de fichiers, car les suppressions seront d’un côté, alors que les ajouts seront d’un autre côté. Ce qui ne sera pas modifié sera inchangé des deux côtés.
> Quand vous remarquez une faute, allez sur la ligne qui vous intéresse (le numéro des lignes est à gauche, dans la marge), vous verrez un petit [b]+[/b] apparaître, cliquez dessus.
> Une fenêtre de dialogue va s’afficher, avec une zone de texte. Nous allons aller sur la barre d’outils en haut de cette zone de texte, et nous allons cliquer sur [b][url=https://imgur.com/BjXWtGj.png]ce bouton[/url][/b].

Author: ```suggestion
Je vais rester sur la Pull Request que je viens de créer, qui servira d’exemple dans ce processus.
Lisez attentivement l’article pour trouver de potentielles fautes.

Remarque : dans les Pull Requests qui modifient des fichiers déjà existants, concentrez-vous sur les parties en vert, et non celles en rouge, qui correspondent à ce qui a été enlevé. Si vous avez des difficultés, vous pouvez changer la vue de cet onglet en haut, en cliquant sur l’engrenage au milieu en haut de la page, et en cliquant sur le bouton [b]Split[/b], puis, cliquez sur [b]Apply and reload[/b]. 
Cette vue simplifiera la lecture des modifications pour ce type de fichiers, car les suppressions seront d’un côté, alors que les ajouts seront d’un autre côté. Ce qui ne sera pas modifié sera inchangé des deux côtés.

Quand vous remarquez une faute, allez sur la ligne qui vous intéresse (le numéro des lignes est à gauche, dans la marge), vous verrez un petit [b]+[/b] apparaître, cliquez dessus.
Une fenêtre de dialogue va s’afficher, avec une zone de texte. Nous allons aller sur la barre d’outils en haut de cette zone de texte, et nous allons cliquer sur [b][url=https://imgur.com/BjXWtGj.png]ce bouton[/url][/b].
```

La ligne que vous avez sélectionnée sera entourée par une fonction [b]suggestion[/b]. Dans cette espace, modifiez ce qui doit être modifié. Quand vous aurez terminé, vous avez deux choix :
[list]
[*]Un bouton [b]Add single comment[/b] qui va directement poster la suggestion

> [*]Un bouton [b]Start a review[/b] qui va mettre votre suggestion en attente, et vous permettre de proposer d’autres modifications si vous en avez découvert.

Author: ```suggestion
[*]Un bouton [b]Start a review[/b] qui va mettre votre suggestion en attente, et vous permettre de proposer d’autres modifications si vous en avez décelé plusieurs (conseillé car cela est mieux de poster toutes les suggestions d'un seul coup que de les poster une par une).
```

Si vous estimez que la Pull Request est correcte, au lieu de sélectionner [b]Request changes[/b], cliquez sur [b]Approve[/b] pour approuver les modifications proposées par ce traducteur.

Nous avons fait le tour de tout ce que vous devez savoir sur le processus de traduction. Cela a été un peu long, mais il est important de détailler tout le processus dans son ensemble, afin que vous puissiez comprendre autant que possible ce processus qui reste complexe sans des explications claires et précises.

[b][size=150]FAQ[/size][/b]

[b]Question :[/b] J'ai créé une Pull Request de mes modifications, un contributeur français à approuvé les modifications que j'ai faites, et maintenant ?

[b]Réponse :[/b] Si un contributeur français a approuvé vos modifications sur votre Pull Request, vous pouvez soit attendre qu'un maintainer fusionne votre Pull Request (on appelle ça aussi [b]Merge[/b]), soit demander à ces personnes de le faire en leur envoyant un message privé sur leur Discord (voir "liens utiles" en fin de post).

> [b]Question :[/b] On me dit que ma branche est [i]"obsolète"[/i], qu'est-ce que cela signifie ?

Author: ```suggestion
[b]Question :[/b] On me dit que ma branche est [i]"obsolète"[/i] ou j'ai l'impression qu'elle l'est, qu'est-ce que cela signifie ?
```

[b]Réponse :[/b] Une branche obsolète est [i]"normale"[/i], car lorsque vous avez cloné le projet sur une version que vous pouvez modifier, vous l'avait fait à un instant T, et cette dernière ne se mettant pas à jour automatiquement, vous devez le faire manuellement.

Pour ce faire : 

[list]
[*]Allez sur votre fork (vous pouvez y accéder en allant sur la page d'accueil du dépôt, puis en cliquant sur [b]Fork[/b] en haut à droite de la page)
[*]Cliquez sur [b]branches[/b] en haut à gauche
[*]Cherchez la branche qui pose problème
[*]Quand vous serez sur votre branche, regardez en haut de la page, en dessous de [b]branches[/b], un message sera écrit.

> Si votre branche a [b][url=https://imgur.com/8wIrVwP.jpg]ce statut[/url][/b], vous n'avez rien à faire : elle est à jour

Author: ```suggestion
- Si votre branche a [b][url=https://imgur.com/8wIrVwP.jpg]ce statut[/url][/b], vous n'avez rien à faire : elle est à jour
```

> Si votre branche a [b][url=https://imgur.com/u9wICMw.jpg]ce statut[/url][/b], vous n'avez rien à faire : ce sont les modifications que vous avez apportées

Author: ```suggestion
- Si votre branche a [b][url=https://imgur.com/u9wICMw.jpg]ce statut[/url][/b], vous n'avez rien à faire : ce sont les modifications que vous avez apportées
```

> Si votre branche a [b][url=https://imgur.com/GVoiq0L.jpg]ce statut[/url][/b], votre branche a du retard sur la branche d'origine, vous devez donc la mettre à jour.

Author: ```suggestion
- Si votre branche a [b][url=https://imgur.com/GVoiq0L.jpg]ce statut[/url][/b], votre branche a du retard sur la branche d'origine, vous devez donc la mettre à jour.
```

> Si votre branche a [b][url=https://imgur.com/oVvkgDM.jpg]ce statut[/url][/b], votre branche a du retard et de l'avance sur la branche d'origine, vous devez la mettre à jour car il y a du retard.

Author: ```suggestion
- Si votre branche a [b][url=https://imgur.com/oVvkgDM.jpg]ce statut[/url][/b], votre branche a du retard et de l'avance sur la branche d'origine, vous devez la mettre à jour car il y a du retard.
```

[*]Si votre statut correspond à l'un des deux qui signifient que votre branche n'est pas à jour, cliquez sur [b][url=https://imgur.com/F3N0zrg.png]ce bouton[/url][/b] qui est à gauche du message, cliquez ensuite sur [b]Fetch and merge[/b] pour mettre à jour votre branche.
[*]Pensez également à mettre à jour de temps en temps votre branche [b]master[/b], c'est sur cette branche que va se baser toutes les autres, vous devez donc essayer de la garder le plus à jour possible.
[/list]
[b]Question :[/b] Après avoir mis à jour ma branche, j'ai des conflits, mais je ne sais pas comment les résoudre !

> [b]Réponse :[/b] Les conflits correspondent à des modifications qui ont été merged avant les vôtres sur les fichiers que vous modifiez, et ce sont ces modifications qui entrent en conflit avec les vôtres.

Author: ```suggestion
[b]Réponse :[/b] Les conflits correspondent à des modifications qui ont été merge avant les vôtres sur les fichiers que vous modifiez, et ce sont ces modifications qui entrent en conflit avec les vôtres.
```

Selon ce que vous souhaitez faire, sélectionner l'une des deux options.

Quand vous reviendrez sur votre branche en ayant sélectionné [b]Leave my changes on {nom de votre branche}[/b], vous avez un [b][url=https://i.imgur.com/QSYgQfe.png]bouton[/url][/b] qui se trouve dans la zone réservée habituellement a l'affichage des changements effectués.
Cliquez sur ce bouton. Vous allez arriver [b][url=https://i.imgur.com/HL0h0ZM.png]ici[/url][/b]. Vous avez deux choix : [b]Restore[/b] pour restaurer les modifications qui s'affiche, [b]Discard[/b] pour abandonner les modifications affichées, elles seront totalement perdues après cela.

[b][size=150]Liens utiles[/size][/b]

[b][url=https://discord.com/invite/vfrHNQZ]Lien vers le Discord de la communauté française d'osu![/url][/b] (un salon #traduction est réservé à la traduction). Vous avez besoin d'un compte osu! afin de rejoindre le Discord, une vérification sera faite par les modérateurs.
[b][url=https://github.com/ppy/osu-wiki]Dépôt du projet osu-wiki[/url][/b]

> [b][url=https://docs.google.com/spreadsheets/d/1zppXOH3PK32c0FwR8TC8QixxzoliEPcWNOjMfAfAv8o/edit#gid=1581693581]Spreadsheet du projet[/url][/b]

Author: ```suggestion
```

[b][size=150]Contribuer au projet de traduction de l’osu! wiki en français[/size][/b]

Bonjour à tous et à toutes !

J’aimerais vous présenter un projet qui a été créé le 3 août 2019 : un vaste projet de traduction de l’osu! wiki en français.

Ce projet a pour but de traduire l'[b][url=https://osu.ppy.sh/wiki/fr/Main_Page]osu! wiki[/url][/b] en français. Ce wiki est destiné à osu!, et réunit toutes les informations qui concernent le jeu.

Nous avons donc besoin de vous, pour construire un wiki compréhensible de toute la communauté française, pour vous, mais aussi et surtout pour les autres.

> Il n’y a pas forcément de prérequis pour rejoindre le projet : un niveau d’anglais et un orthographe corrects sont cependant indispensables pour le comprendre et le restituer en français.

Author: ```suggestion
Il n’y a pas forcément de prérequis pour rejoindre le projet : un niveau d’anglais et un orthographe corrects sont cependant indispensables pour comprendre les contenus en anglais et les restituer en français, si possible sans trop de fautes de français.
```

> Pour vous permettre de contribuer à tout cela facilement, nous avons décidé de réaliser un tutoriel afin de vous apprendre comment vous pouvez contribuer au mieux, correctement, et dans de bonnes conditions de travail.

Author: ```suggestion
Pour vous permettre de contribuer à tout cela facilement, nous avons décidé de réaliser un tutoriel afin de vous apprendre comment vous pouvez contribuer au mieux, correctement, et dans de bonnes conditions de travail.

Ce tutoriel a été écrit et imaginé par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], complété et amélioré par [b][url=https://osu.ppy.sh/users/10660738]Ganondorf[/url][/b].
```

[b][size=150]GitHub[/size][/b]

Toute la contribution du projet se passe sur [b]GitHub[/b], qui est une plateforme sur laquelle presque tous les projets ou logiciel sont basés, car elle permet à ces derniers d’être open-source (en clair, on peut voir la structure d’un logiciel gratuitement et sans risque, car ce seront les développeurs qui l’auront voulu !).

Pour commencer, nous allons vous montrer comment créer un compte sur cette plateforme, et comment mettre en place votre compte GitHub afin de pouvoir accueillir le projet.

[b][size=150]Création du compte[/size][/b]

Pour commencer, nous allons créer un compte. Pour ce faire, dirigez-vous [b][url=https://github.com/signup?ref_cta=Sign+up&ref_loc=header+logged+out&ref_page=%2F&source=header-home]ici[/url][/b], et laissez-vous guider.

Quand votre compte est configuré (et que vous l’aurez peut-être paramétré à votre goût), vous allez pouvoir accéder au « dépôt » du projet. 
Un dépôt (appelé [i]repository[/i]) est l’endroit où vous pouvez consulter le projet dans sa globalité, proposer des changements, ou encore signaler des problèmes. 
Mais chaque chose en son temps, nous aurons le temps de voir tout cela plus tard.

> Donc, pour pouvoir accéder au dépôt du projet, dirigez-vous [b][url=https://github.com/ppy/osu-wiki]ici[/url][/b]. Si tout va bien, vous allez tomber sur une page qui ressemble à [b][url=https://i.imgur.com/bhBMMSG.png]celle-ci[/url][/b].

Author: ```suggestion
Donc, pour pouvoir accéder au dépôt du projet, dirigez-vous [b][url=https://github.com/ppy/osu-wiki]ici[/url][/b]. Si tout va bien, vous allez tomber sur une page qui ressemble à [b][url=https://imgur.com/bhBMMSG.png]celle-ci[/url][/b].
```

Dès que vous êtes ici, vous êtes sur le projet en lui-même : c'est ici que tout va se faire.
Avant de pouvoir traduire votre première page, vous devez faire une copie du projet que vous pourrez modifier sans risque.

> Pour ce faire, vous allez repérer les trois boutons en haut à droite de votre page, et cliquer sur [b][url=https://i.imgur.com/IywtwKK.png]celui-ci[/url][/b]. « Fork » (nom barbare en premier lieu) veut tout simplement dire que cela va copier le projet dans son intégralité, en une version modifiable sans risque.

Author: ```suggestion
Pour ce faire, vous allez repérer les trois boutons en haut à droite de votre page, et cliquer sur [b][url=https://imgur.com/IywtwKK.png]celui-ci[/url][/b]. « Fork » (nom barbare en premier lieu) veut tout simplement dire que cela va copier le projet dans son intégralité, en une version modifiable sans risque.
```

À partir de ce moment, le projet va se copier. Cette copie peut prendre du temps : n’ayez crainte ! Cela est tout à fait normal, le projet est assez volumineux et prend donc un certain temps à se copier.

> Une fois le projet copié sur votre compte, cela devrait ressembler à [b][url=https://i.imgur.com/7lzNZmV.png]ceci[/url][/b]. En premier lieu, vous allez vous dire que rien n’a changé, mais en réalité le projet est désormais modifiable sur un dépôt qui vous est propre, comme si vous en étiez le propriétaire alors qu’il ne s'agit que d'une simple copie de ce projet.

Author: ```suggestion
Une fois le projet copié sur votre compte, cela devrait ressembler à [b][url=https://imgur.com/7lzNZmV.png]ceci[/url][/b]. En premier lieu, vous allez vous dire que rien n’a changé, mais en réalité le projet est désormais modifiable sur un dépôt qui vous est propre, comme si vous en étiez le propriétaire alors qu’il ne s'agit que d'une simple copie de ce projet.
```

[b][size=150]GitHub Desktop[/size][/b]

Maintenant que vous avez cette copie, nous allons pouvoir parler d’un logiciel très important : [b]GitHub Desktop[/b].

Qu’est-ce que GitHub Desktop ? Il s'agit d'un logiciel (créé par GitHub) totalement gratuit qui va vous permettre de lier GitHub avec un éditeur de texte (que nous verrons plus tard), afin de pouvoir déposer vos traductions sur le projet. Il est indispensable, car sans lui, ce processus serait compliqué !

> Pour pouvoir télécharger GitHub Desktop, dirigez-vous [b][url=https://desktop.github.com/]ici[/url][/b], et cliquez [b][url=https://i.imgur.com/veRq4jI.png]ici[/url][/b] pour le télécharger.

Author: ```suggestion
Pour pouvoir télécharger GitHub Desktop, dirigez-vous [b][url=https://desktop.github.com/]ici[/url][/b], et cliquez [b][url=https://imgur.com/veRq4jI.png]ici[/url][/b] pour le télécharger.
```

Par la suite, exécutez le fichier sur votre ordinateur, et installez-le comme un logiciel classique.

> Une fois téléchargé, il va vous proposer [b][url=https://i.imgur.com/jLQMOvt.png]de vous connecter[/url][/b], sélectionnez l’option [b]Sign in to GitHub.com[/b].

Author: ```suggestion
Une fois téléchargé, il va vous proposer [b][url=https://imgur.com/jLQMOvt.png]de vous connecter[/url][/b], sélectionnez l’option [b]Sign in to GitHub.com[/b].
```

> GitHub Desktop va alors vous proposer de [b][url=https://i.imgur.com/IiLGTEk.png]configurer votre adresse mail[/url][/b], je vous conseille de la laisser comme telle, à moins que vous vouliez laisser apparaître l’email avec laquelle vous avez créé votre compte publiquement. Cliquez donc sur [b]Finish[/b].

Author: ```suggestion
GitHub Desktop va alors vous proposer de [b][url=https://imgur.com/IiLGTEk.png]configurer votre adresse mail[/url][/b], je vous conseille de la laisser comme telle, à moins que vous vouliez laisser apparaître l’email avec laquelle vous avez créé votre compte publiquement. Cliquez donc sur [b]Finish[/b].
```

> Enfin, il va vous proposer [b][url=https://i.imgur.com/gVZPZyn.png]de cloner le dépôt sur votre ordinateur[/url][/b]. Si tout va bien, vous aurez à droite la liste des dépôts qui sont présents sur votre compte. Cliquez sur [b]{votre nom d’utilisateur}/osu-wiki[/b], puis sur [b]Clone{votre nom d’utilisateur}/osu-wiki[/b], et une nouvelle fois sur [b]Clone[/b].

Author: ```suggestion
Enfin, il va vous proposer [b][url=https://imgur.com/gVZPZyn.png]de cloner le dépôt sur votre ordinateur[/url][/b]. Si tout va bien, vous aurez à droite la liste des dépôts qui sont présents sur votre compte. Cliquez sur [b]{votre nom d’utilisateur}/osu-wiki[/b], puis sur [b]Clone{votre nom d’utilisateur}/osu-wiki[/b], et une nouvelle fois sur [b]Clone[/b].
```

> Si rien ne s'affiche, cliquez sur le bouton [b]Clone a repository from the Internet…[/b], [b][url=https://i.imgur.com/GVtkqIZ.png]une petite fenêtre va s’ouvrir[/url][/b], si vous ne voyez toujours rien, allez dans la section [b]URL[/b], et tapez [b]{votre nom d’utilisateur}/osu-wiki[/b].

Author: ```suggestion
Si rien ne s'affiche, cliquez sur le bouton [b]Clone a repository from the Internet…[/b], [b][url=https://imgur.com/GVtkqIZ.png]une petite fenêtre va s’ouvrir[/url][/b], si vous ne voyez toujours rien, allez dans la section [b]URL[/b], et tapez [b]{votre nom d’utilisateur}/osu-wiki[/b].
```

Si tout va bien, le projet devrait se télécharger sur votre ordinateur. Cela sera également long, car le projet est volumineux.

> Lorsque le projet aura été copié sur votre ordinateur, vous devriez avoir une fenêtre qui ressemble à [b][url=https://i.imgur.com/Sz5CYFB.png]celle-ci[/url][/b]. Cette fenêtre va vous demander comment vous comptez utiliser votre fork.

Author: ```suggestion
Lorsque le projet aura été copié sur votre ordinateur, vous devriez avoir une fenêtre qui ressemble à [b][url=https://imgur.com/Sz5CYFB.png]celle-ci[/url][/b]. Cette fenêtre va vous demander comment vous comptez utiliser votre fork.
```

[list]
[*][b]To contribute to the parent project[/b] pour contribuer sur le projet « parent », il s'agira dans notre cas de [b]ppy/osu-wiki[/b]
[*][b]For my own purposes[/b] pour travailler sur votre propre fork
[/list]

Puisque nous voulons partager nos traductions, je vous conseille de cliquer sur [b]To contribute to the parent project[/b].

> Après toutes ces étapes, vous y êtes ! voici [b][url=https://i.imgur.com/nPPkZp7.png]la fenêtre principale de GitHub Desktop[/url][/b].

Author: ```suggestion
Après toutes ces étapes, vous y êtes ! voici [b][url=https://imgur.com/nPPkZp7.png]la fenêtre principale de GitHub Desktop[/url][/b].
```

Les étapes sont plutôt longues, mais nous essayons de détailler le plus possible afin que tout le monde puisse s’y retrouver.
Pour le moment, nous allons laisser de côté GitHub Desktop, et nous allons commencer la mise en place de l’éditeur de texte.

[b][size=150]Visual Studio Code[/size][/b]

Il existe différents logiciels de traitement de texte fonctionnant avec GitHub Desktop. Sur ce tutoriel, nous avons décidé d’utiliser [b]Visual Studio Code[/b], qui est selon nous, le plus simple d’utilisation.

Visual Studio Code, c’est un logiciel de traitement de texte comme il en existe des milliers. Sauf que ce dernier est plus destiné à l’écriture de lignes de codes et au développement. C’est comme [b]Notepad++[/b] pour les connaisseurs, mais avec des possibilités presque infinies.
Comme GitHub Desktop, Visual Studio Code est gratuit, et est [i]normalement[/i] plus simple d’utilisation.

Pour commencer, [b][url=https://code.visualstudio.com/]installez le logiciel[/url][/b]. Installez-le comme un logiciel classique, tout comme GitHub Desktop.

> Ensuite, quand tout est installé, vous devriez tomber sur une fenêtre comme [b][url=https://i.imgur.com/qvyqr7Y.png]celle-ci[/url][/b], c’est la fenêtre principale de Visual Studio Code.

Author: ```suggestion
Ensuite, quand tout est installé, vous devriez tomber sur une fenêtre comme [b][url=https://imgur.com/qvyqr7Y.png]celle-ci[/url][/b], c’est la fenêtre principale de Visual Studio Code.
```

Pour utiliser le logiciel dans le cadre de la contribution à l'osu! wiki, vous devez installer des extensions, permettant d’ajouter des fonctionnalités à l’éditeur.

> Pour installer des extensions, vous devez cliquer sur [b][url=https://i.imgur.com/kCgmJ2l.png]ce bouton[/url][/b] pour accéder au menu des extensions.

Author: ```suggestion
Pour installer des extensions, vous devez cliquer sur [b][url=https://imgur.com/kCgmJ2l.png]ce bouton[/url][/b] pour accéder au menu des extensions.
```

> Vous allez avoir une barre latérale sur votre gauche comme [b][url=https://i.imgur.com/I4Mc8Br.png]celle-ci[/url][/b].

Author: ```suggestion
Vous allez avoir une barre latérale sur votre gauche comme [b][url=https://imgur.com/I4Mc8Br.png]celle-ci[/url][/b].
```

En haut de cette barre latérale, vous avez une barre de recherche, sur cette dernière, vous allez installer deux extensions, indispensables pour la contribution : [b][url=https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens]GitLens — Git supercharged[/url][/b] de [b]Eric Amodio[/b] et [b][url=https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one]Markdown All in One[/url][/b] de [b]Yu Zhang[/b].

GitLens est un outil permettant d’inclure toutes les fonctions de GitHub et de Git (voyez ça comme un « plugin ») dans Visual Studio Code.

Markdown All in One est un outil permettant d'inclure toutes les fonctions du langage « Markdown » dans Visual Studio Code. Markdown se rapproche des langages MediaWiki ou BBCode, mais avec plus de possibilités et de simplicité.

Maintenant que tout est configuré, vous allez [i]enfin[/i] pouvoir commencer la traduction à proprement parler.

Avant cela, une dernière petite chose à faire : vous devez sélectionner le dossier où se situe le projet et l’ouvrir dans Visual Studio Code.

> Pour ce faire, cliquez sur [b][url=https://i.imgur.com/FUiRZ4B.png]cette icône[/url][/b] pour revenir à l’explorateur de fichiers, et cliquez sur le bouton [b]Ouvrir un dossier[/b], puis cherchez le dossier du projet, il s’appelle [b]osu-wiki[/b].

Author: ```suggestion
Pour ce faire, cliquez sur [b][url=https://imgur.com/FUiRZ4B.png]cette icône[/url][/b] pour revenir à l’explorateur de fichiers, et cliquez sur le bouton [b]Ouvrir un dossier[/b], puis cherchez le dossier du projet, il s’appelle [b]osu-wiki[/b].
```

> Si vous ne savez pas où se trouve ce dossier, retournez sur [b]GitHub Desktop[/b], et cliquez [b][url=https://i.imgur.com/qhudFRh.png]sur ce bouton[/url][/b], puis cliquez droit sur [b]osu-wiki[/b], et sélectionnez [b]Show in Explorer[/b]. Maintenant que vous savez où se trouve le dossier, refaites les étapes plus haut pour l’ouvrir dans Visual Studio Code.

Author: ```suggestion
Si vous ne savez pas où se trouve ce dossier, retournez sur [b]GitHub Desktop[/b], et cliquez [b][url=https://imgur.com/qhudFRh.png]sur ce bouton[/url][/b], puis cliquez droit sur [b]osu-wiki[/b], et sélectionnez [b]Show in Explorer[/b]. Maintenant que vous savez où se trouve le dossier, refaites les étapes plus haut pour l’ouvrir dans Visual Studio Code.
```

> Une fois le dossier ouvert, [b][url=https://i.imgur.com/AIzofdh.png]voici à quoi ressemble l’explorateur[/url][/b].

Author: ```suggestion
Une fois le dossier ouvert, [b][url=https://imgur.com/AIzofdh.png]voici à quoi ressemble l’explorateur[/url][/b].
```

[b]Une fois le dossier ouvert, j’aimerais vous parler de quelques règles et vous donner quelques conseils concernant la traduction, afin que vous réalisiez cette dernière de la meilleure façon possible :[/b]

[list]
[*]GitHub a mis en place un système de [b]branches[/b] (branch en anglais), permettant d’inclure des modifications sur plusieurs de ces branches, ce qui vous permet de traduire plusieurs fichiers différents, sans que cela puisse vous gêner avec la précédente.

> Pour créer une branche, vous devez aller dans GitHub Desktop, et cliquer sur [b][url=https://i.imgur.com/t7i3OFZ.png]ce bouton[/url][/b], qui va vous ouvrir [b][url=https://i.imgur.com/zvgusd1.png]une liste déroulante comme celle-ci[/url][/b] avec toutes les branches que vous possédez (vous n’en aurez aucune au départ).

Author: ```suggestion
Pour créer une branche, vous devez aller dans GitHub Desktop, et cliquer sur [b][url=https://imgur.com/t7i3OFZ.png]ce bouton[/url][/b], qui va vous ouvrir [b][url=https://imgur.com/zvgusd1.png]une liste déroulante comme celle-ci[/url][/b] avec toutes les branches que vous possédez (vous n’en aurez aucune au départ).
```

Pour en créer une, cliquez sur [b]New branch[/b] en haut à droite, mettez ensuite le nom de votre branche (nous vous conseillons de mettre le nom de la page que vous souhaitez traduire (en remplaçant les espaces par des [b]-[/b] comme le nom ne doit pas contenir d’espaces, sinon il le fera pour vous), avec juste après la mention [b]FR[/b] pour signifier que c’est une traduction en langue française.
Lorsque vous aurez nommé votre branche, cliquez sur le bouton [b]Create branch[/b], et le tour est joué !

> [*]Tout les articles n’ont pas forcément besoin d’être traduits/mis à jour, c’est pour cela qu’un modérateur du projet a décidé de mettre en place [b][url=https://clayton.cc/osu-wiki-status/fr]ce site[/url][/b] qui recense toutes les pages nécessitant notre attention par catégorie. Vous devez éviter de traduire les pages qui se trouvent dans les catégories [b]Missing articles (outdated in English)[/b], ou toutes celles qui possèdent [b][url=https://i.imgur.com/6KSQCmB.png]ceci[/url][/b] et/ou [b][url=https://i.imgur.com/46EmsAQ.png]cela[/url][/b] sur le fichier [b]en.md[/b] (qui est le fichier en langue anglaise du fichier, celui sur lequel vous allez vous baser pour traduire)

Author: ```suggestion
[*]Tout les articles n’ont pas forcément besoin d’être traduits/mis à jour, c’est pour cela qu’un modérateur du projet a décidé de mettre en place [b][url=https://clayton.cc/osu-wiki-status/fr]ce site[/url][/b] qui recense toutes les pages nécessitant notre attention par catégorie. Vous devez éviter de traduire les pages qui se trouvent dans les catégories [b]Missing articles (outdated in English)[/b], ou toutes celles qui possèdent [b][url=https://imgur.com/6KSQCmB.png]ceci[/url][/b] et/ou [b][url=https://imgur.com/46EmsAQ.png]cela[/url][/b] sur le fichier [b]en.md[/b] (qui est le fichier en langue anglaise du fichier, celui sur lequel vous allez vous baser pour traduire)
```

Le tag [b]outdated: true[/b] signifie que l’article n’est plus à jour, et qu'il peut à tout moment être modifié en ajoutant de nouvelles informations.
Le tag [b]need_cleanup: true[/b] signifie que l’article a besoin de nettoyage, soit d’informations superflues, soit au niveau de la mise en forme/écriture de l’article.
En complément de cela, vous avez [b][url=https://github.com/ppy/osu-wiki/issues/2486]ces explications[/url][/b] sur ce qui a été dit.

[*]Avant de choisir votre page, vous devez vous assurer que cette dernière n’est pas déjà en train d’être traduite par un autre contributeur. C’est pour cela que nous avons mis en place [b][url=https://docs.google.com/spreadsheets/d/1zppXOH3PK32c0FwR8TC8QixxzoliEPcWNOjMfAfAv8o/edit#gid=1581693581]cette spreadsheet[/url][/b] qui liste les articles qui sont en cours de traduction, ceux qui sont en attente de vérification par d’autres traducteurs, ainsi que la liste (informative) des traductions terminées et publiées sur le wiki.

[*]Quand vous aurez choisi votre page, vous pouvez tout traduire, sauf :

> Ce qui se trouve entre parenthèses avec un lien comme [b][url=https://i.imgur.com/GUmoQ38.png]celui-ci[/url][/b]. Ce sont des liens de redirections vers d’autres articles, vous ne devez pas y toucher, sinon la redirection ne fonctionnera plus.

Author: ```suggestion
Ce qui se trouve entre parenthèses avec un lien comme [b][url=https://imgur.com/GUmoQ38.png]celui-ci[/url][/b]. Ce sont des liens de redirections vers d’autres articles, vous ne devez pas y toucher, sinon la redirection ne fonctionnera plus.
```

Les termes spécifiques au jeu qui n’ont pas de traduction française (se référer à [b][url=https://gist.github.com/DoryanR/59b20e0fb40f27a371761f2fd002cccc]cette liste[/url][/b] si vous avez un doute).
Sachez que votre traduction ne sera pas tout de suite mise sur le site : elle devra d'abord être vérifiée par d’autres traducteurs français avant qu’elle soit mise sur le site, n’ayez pas peur de faire des erreurs !

[*]Quand vous devez créer un fichier pour accueillir la traduction française d’une page, appelez votre fichier [b]fr.md[/b].

[*]Un conseil : quand vous souhaitez traduire une page, je vous conseille de prendre tout le contenu du fichier original en anglais, de le copier sur votre fichier de traduction, et de traduire directement dessus. Ça vous évite de devoir refaire toute la mise en forme, et c’est plus rapide.
[/list]

Maintenant que vous savez tout ça, vous pouvez vous lancer.

[b][size=150]Mettre en ligne la contribution[/size][/b]

> Une fois votre traduction terminée, revenez sur GitHub Desktop, la fenêtre principale [b][url=https://i.imgur.com/QqLYndG.png]aura légèrement changé[/url][/b].

Author: ```suggestion
Une fois votre traduction terminée, revenez sur GitHub Desktop, la fenêtre principale [b][url=https://imgur.com/QqLYndG.png]aura légèrement changé[/url][/b].
```

Sur la gauche, vous avez la liste des fichiers que vous avez créés, supprimés et modifiés. Un [b]+[/b] vert apparaîtra à côté du nom du fichier s'il s'agit d'un fichier créé, un [b].[/b] orange s'il s'agit d'un fichier modifié, et un [b]-[/b] rouge s'il s'agit d'un fichier supprimé.

Au milieu, vous avez le contenu du fichier, avec en vert tout ce que vous aurez ajouté, et en rouge tout ce qui aura été enlevé.
Avant de la mettre en ligne, vous devez créer un [b]commit[/b] de votre modification.
Un commit est un terme qu’on utilise pour désigner une modification que l’on apporte à une branche, par rapport à la version qui est déjà sur le projet.

> Pour créer ce commit, vous devez vous diriger en bas à gauche de la fenêtre de GitHub Desktop, plus particulièrement [b][url=https://i.imgur.com/fVRnAwB.png]ici[/url][/b].

Author: ```suggestion
Pour créer ce commit, vous devez vous diriger en bas à gauche de la fenêtre de GitHub Desktop, plus particulièrement [b][url=https://imgur.com/fVRnAwB.png]ici[/url][/b].
```

Dans la barre à côté de votre avatar, vous devez spécifier un titre à votre commit. Nous vous conseillons de mettre ce genre de titre : [b][FR]{Add}/{Update} {`Nom de l’article`}[/b].
Dans mon cas, je vais mettre le titre suivant : [b][FR] Add `Antiflow`[/b], car j’ai ajouté une page, et j’ai ajouté la traduction de l’article [b]Antiflow[/b].

Dans la zone de texte plus bas, vous pouvez ajouter si vous le souhaitez une description de ce que vous avez fait (en anglais), mais cela n’est pas obligatoire.

> Quand tout est bon, vous pouvez cliquer sur [b]Commit to {nom de votre branche}[/b]. Vous verrez apparaître [url=https://i.imgur.com/nXB4upz.png]ceci[/url]. Cela veut dire que votre commit a bien fonctionné, et qu’elle est sur votre branche, elle est alors en attente.

Author: ```suggestion
Quand tout est bon, vous pouvez cliquer sur [b]Commit to {nom de votre branche}[/b]. Vous verrez apparaître [url=https://imgur.com/nXB4upz.png]ceci[/url]. Cela veut dire que votre commit a bien fonctionné, et qu’elle est sur votre branche, elle est alors en attente.
```

Puisqu'il s'agit de votre premier commit, vous n’avez pas de [b]Pull Request[/b] active, vous devez donc en créer une.
Une Pull Request (PR en abrégé), c’est l’endroit où vous allez proposer votre modification sur le projet, pour qu’elle soit critiquée et qu’on vous propose des modifications. Le cas échéant, vos modifications seront approuvées si nous n’avons rien à redire.

> Pour en créer une, nous resterons sur GitHub Desktop, et remarquer que la fenêtre d’origine a [b][url=https://i.imgur.com/JoQKxOC.png]légèrement changé[/url][/b].

Author: ```suggestion
Pour en créer une, nous resterons sur GitHub Desktop, et remarquer que la fenêtre d’origine a [b][url=https://imgur.com/JoQKxOC.png]légèrement changé[/url][/b].
```

Comme vous pouvez le voir, un message bleu vous dira ce qui a été dit plus haut : que votre branche n’est pas publiée sur GitHub, et n’a donc pas de Pull Request à son actif. Pour la créer, vous pouvez cliquer sur [b]Publish Branch[/b] à droite du message, ou cliquer en haut sur [b]Publish Branch[/b], juste à côté de l'endroit où vous avez créé votre branche. Les deux actions amènent au même résultat.
Une fois que vous aurez cliqué sur l’un des deux boutons, votre branche sera être publiée (celle s'appelle un [b]push[/b]) sur GitHub.

Désormais, vous pouvez revenir sur [b][url=https://github.com/ppy/osu-wiki]l’accueil du dépôt[/url][/b], ou rester sur GitHub Desktop et cliquer sur [b]Create Pull Request[/b] au centre de la fenêtre, au même endroit où il vous propose de publier la branche sur GitHub.

> Si vous choisissez de retourner sur le dépôt, vous pourrez observer un [b][url=https://i.imgur.com/1sSHcQZ.png]petit changement[/url][/b] en haut de votre écran. Cette fenêtre en orange est la branche que vous venez juste de publier sur GitHub, cliquez donc sur [b]Compare & pull request[/b].

Author: ```suggestion
Si vous choisissez de retourner sur le dépôt, vous pourrez observer un [b][url=https://imgur.com/1sSHcQZ.png]petit changement[/url][/b] en haut de votre écran. Cette fenêtre en orange est la branche que vous venez juste de publier sur GitHub, cliquez donc sur [b]Compare & pull request[/b].
```

> Vous allez arriver sur [b][url=https://i.imgur.com/9ENAr8z.png]cette fenêtre[/url][/b]. C’est là que vous pouvez voir les informations sur votre branche, avec les modifications que vous voulez proposer.

Author: ```suggestion
Vous allez arriver sur [b][url=https://imgur.com/9ENAr8z.png]cette fenêtre[/url][/b]. C’est là que vous pouvez voir les informations sur votre branche, avec les modifications que vous voulez proposer.
```

Ce qui se trouve tout en haut, ne vous en occupez pas, ce sont simplement les paramètres qui définissent sur quel dépôt du projet il doit se baser pour qu’il soit publié par la suite.

Juste en dessous, vous avez le titre de votre premier commit, qui est automatiquement transformé en votre titre pour votre Pull Request, avec la description en dessous (si vous en avez ajouté une).
Si vous allez en dessous, vous avez le même affichage que tout à l’heure sur GitHub Desktop, avec ce que vous avez ajouté au fichier (ou supprimé).

> Une petite information importante : en bas de la zone de texte pour la description, vous avez [b][url=https://i.imgur.com/1ucKWCA.png]ceci[/url][/b]. Ce paramètre doit être coché, car il permet aux modérateurs du projet (ils sont appelés [b]Maintainers[/b]) d’accéder à votre branche et d’y apporter des modifications, comme ce que vous ferez plus tard. Cela n’est pas gênant de leur laisser l’accès à votre branche, car seuls eux pourront y accéder.

Author: ```suggestion
Une petite information importante : en bas de la zone de texte pour la description, vous avez [b][url=https://imgur.com/1ucKWCA.png]ceci[/url][/b]. Ce paramètre doit être coché, car il permet aux modérateurs du projet (ils sont appelés [b]Maintainers[/b]) d’accéder à votre branche et d’y apporter des modifications, comme ce que vous ferez plus tard. Cela n’est pas gênant de leur laisser l’accès à votre branche, car seuls eux pourront y accéder.
```

Lorsque vous aurez ajouté une description (vous pouvez simplement dire que vous avez traduit tel article, cela suffira), cliquez sur [b]Create pull request[/b].

> Une fois votre Pull Request publiée, voici [b][url=https://i.imgur.com/VsIAK7L.png]à quoi cela ressemble[/url][/b].

Author: ```suggestion
Une fois votre Pull Request publiée, voici [b][url=https://imgur.com/VsIAK7L.png]à quoi cela ressemble[/url][/b].
```

À partir de ce moment, vous avez fait le plus dur. Vous pouvez désormais attendre que d’autres traducteurs français regardent votre Pull Request, y apportent des suggestions, et que les modifications soient approuvées, vous pouvez pendant ce temps commencer à traduire un autre article.

[b][size=150]Apporter des modifications à ma Pull Request[/size][/b]

Des modifications ont été proposées sur votre Pull Request ? Voici comment les accepter, ou proposer des modifications si elles vous semblent incorrectes.

Si l’un des traducteurs propose des modifications sur votre Pull Request, vous devez soit appliquer les modifications que l’on vous a proposées, soit les commenter si vous n’êtes pas d’accord.
Dans les deux cas, tout se passe dans l’onglet [b]Files changed[/b].

> L’onglet ressemble à [b][url=https://i.imgur.com/YIUOdib.png]ceci[/url][/b].

Author: ```suggestion
L’onglet ressemble à [b][url=https://imgur.com/YIUOdib.png]ceci[/url][/b].
```

Pour faire un exemple, je vais simuler une demande de modification de ma propre Pull Request (c’est possible, et cela est plus simple que de repasser par Visual Studio Code pour apporter des modifications – nous verrons cela un peu plus tard).

Imaginons que je souhaite mettre une majuscule au mot [b]beatmapping[/b], qui se trouve à la ligne 7.

> [b][url=https://i.imgur.com/IQ7JDku.png]Voici a quoi ressemblera la suggestion[/url][/b].

Author: ```suggestion
[b][url=https://imgur.com/IQ7JDku.png]Voici a quoi ressemblera la suggestion[/url][/b].
```

La partie rouge s'agit de ce qui se trouvait avant la proposition (ce qui est souligné est l’élément sur lequel la personne qui propose une modification agit, ici c’est [b]Beatmapping[/b].

La partie verte est la version qui sera adoptée après acceptation des modifications.

> Vous pouvez voir[b][url=https://i.imgur.com/eCuinvX.png]ici[/url][/b] les modifications sur votre Pull Request, avec deux boutons : [b]Commit suggestion[/b] et [b]Add suggestion to batch[/b].

Author: ```suggestion
Vous pouvez voir[b][url=https://imgur.com/eCuinvX.png]ici[/url][/b] les modifications sur votre Pull Request, avec deux boutons : [b]Commit suggestion[/b] et [b]Add suggestion to batch[/b].
```

Pour proposer des modifications sur la Pull Request d’un autre traducteur, commencez par ouvrir la page de sa Pull Request en cliquant dessus dans l’onglet [b]Pull requests[/b].
Dirigez-vous ensuite dans l’onglet [b]Files changed[/b].

Je vais rester sur la Pull Request que je viens de créer, qui servira d’exemple dans ce processus.
Lisez attentivement l’article pour trouver de potentielles fautes.
Remarque : dans les Pull Requests qui modifient des fichiers déjà existants, concentrez-vous sur les parties en vert, et non celles en rouge, qui correspondent à ce qui a été enlevé. Si vous avez des difficultés, vous pouvez changer la vue de cet onglet en haut, en cliquant sur l’engrenage au milieu en haut, et en appuyant sur le bouton [b]Split[/b]. 

Cette vue simplifiera la lecture des modifications pour ce type de fichiers, car les suppressions seront d’un côté, alors que les ajouts seront d’un autre côté. Ce qui ne sera pas modifié sera inchangé des deux côtés.
Quand vous remarquez une faute, allez sur la ligne qui vous intéresse (le numéro des lignes est à gauche, dans la marge), vous verrez un petit [b]+[/b] apparaître, cliquez dessus.

> Une fenêtre de dialogue va s’afficher, avec une zone de texte. Nous allons aller sur la barre d’outils en haut de cette zone de texte, et nous allons cliquer sur [b][url=https://i.imgur.com/BjXWtGj.png]ce bouton[/url][/b].

Author: ```suggestion
Une fenêtre de dialogue va s’afficher, avec une zone de texte. Nous allons aller sur la barre d’outils en haut de cette zone de texte, et nous allons cliquer sur [b][url=https://imgur.com/BjXWtGj.png]ce bouton[/url][/b].
```

La ligne que vous avez sélectionnée sera entourée par une fonction [b]suggestion[/b]. Dans cette espace, modifiez ce qui doit être modifié. Quand vous aurez terminé, vous avez deux choix :
[list]
[*]Un bouton [b]Add single comment[/b] qui va directement poster la suggestion
[*]Un bouton [b]Start a review[/b] qui va mettre votre suggestion en attente, et vous permettre de proposer d’autres modifications si vous en avez découvert.
[/list]

> Si vous avez terminé votre review avec la fonction [b]Start a review[/b], retournez en haut de la page, et cliquez sur [b][url=https://i.imgur.com/BafVZ2D.png]ce bouton[/url][/b]. Sélectionnez alors la case [b]Request changes[/b] pour indiquer que votre commentaire porte sur la suggestion de modifications sur la Pull Request.

Author: ```suggestion
Si vous avez terminé votre review avec la fonction [b]Start a review[/b], retournez en haut de la page, et cliquez sur [b][url=https://imgur.com/BafVZ2D.png]ce bouton[/url][/b]. Sélectionnez alors la case [b]Request changes[/b] pour indiquer que votre commentaire porte sur la suggestion de modifications sur la Pull Request.
```

Si vous estimez que la Pull Request est correcte, au lieu de sélectionner [b]Request changes[/b], cliquez sur [b]Approve[/b] pour approuver les modifications proposées par ce traducteur.

Nous avons fait le tour de tout ce que vous devez savoir sur le processus de traduction. Cela a été un peu long, mais il est important de détailler tout le processus dans son ensemble, afin que vous puissiez comprendre autant que possible ce processus qui reste complexe sans des explications claires et précises.

> Ce tutoriel a été écrit et imaginé par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], complété et amélioré par [b][url=https://osu.ppy.sh/users/10660738]Ganondorf[/url][/b].

Author: ```suggestion
```

[b]Réponse :[/b] Si un contributeur français a approuvé vos modifications sur votre Pull Request, vous pouvez soit attendre qu'un maintainer fusionne votre Pull Request (on appelle ça aussi [b]Merge[/b]), soit demander à ces personnes de le faire en leur envoyant un message privé sur leur Discord (voir "liens utiles" en fin de post).

[b]Question :[/b] On me dit que ma branche est [i]"obsolète"[/i], qu'est-ce que cela signifie ?

[b]Réponse :[/b] Une branche obsolète est [i]"normale"[/i], car lorsque vous avez cloné le projet sur une version que vous pouvez modifier, vous l'avait fait à un instant T, et cette dernière ne se mettant pas à jour automatiquement, vous devez le faire manuellement.

Pour ce faire : 

[list]
[*]Allez sur votre fork (vous pouvez y accéder en allant sur la page d'accueil du dépôt, puis en cliquant sur [b]Fork[/b] en haut à droite de la page)
[*]Cliquez sur [b]branches[/b] en haut à gauche
[*]Cherchez la branche qui pose problème
[*]Quand vous serez sur votre branche, regardez en haut de la page, en dessous de [b]branches[/b], un message sera écrit.

> Si votre branche a [b][url=https://i.imgur.com/8wIrVwP.jpg]ce statut[/url][/b], vous n'avez rien à faire : elle est à jour

Author: ```suggestion
Si votre branche a [b][url=https://imgur.com/8wIrVwP.jpg]ce statut[/url][/b], vous n'avez rien à faire : elle est à jour
```

> Si votre branche a [b][url=https://i.imgur.com/u9wICMw.jpg]ce statut[/url][/b], vous n'avez rien à faire : ce sont les modifications que vous avez apportées

Author: ```suggestion
Si votre branche a [b][url=https://imgur.com/u9wICMw.jpg]ce statut[/url][/b], vous n'avez rien à faire : ce sont les modifications que vous avez apportées
```

> Si votre branche a [b][url=https://i.imgur.com/GVoiq0L.jpg]ce statut[/url][/b], votre branche a du retard sur la branche d'origine, vous devez donc la mettre à jour.

Author: ```suggestion
Si votre branche a [b][url=https://imgur.com/GVoiq0L.jpg]ce statut[/url][/b], votre branche a du retard sur la branche d'origine, vous devez donc la mettre à jour.
```

> Si votre branche a [b][url=https://i.imgur.com/oVvkgDM.jpg]ce statut[/url][/b], votre branche a du retard et de l'avance sur la branche d'origine, vous devez la mettre à jour car il y a du retard.

Author: ```suggestion
Si votre branche a [b][url=https://imgur.com/oVvkgDM.jpg]ce statut[/url][/b], votre branche a du retard et de l'avance sur la branche d'origine, vous devez la mettre à jour car il y a du retard.
```

> [*]Si votre statut correspond à l'un des deux qui signifient que votre branche n'est pas à jour, cliquez sur [b][url=https://i.imgur.com/F3N0zrg.png]ce bouton[/url][/b] qui est à gauche du message, cliquez ensuite sur [b]Fetch and merge[/b] pour mettre à jour votre branche.

Author: ```suggestion
[*]Si votre statut correspond à l'un des deux qui signifient que votre branche n'est pas à jour, cliquez sur [b][url=https://imgur.com/F3N0zrg.png]ce bouton[/url][/b] qui est à gauche du message, cliquez ensuite sur [b]Fetch and merge[/b] pour mettre à jour votre branche.
```

[*] Si tous les conflits sont résolus, vous pouvez de nouveau cliquer sur le bouton [b]Resolve conflicts[/b].

Si ce bouton est grisé, vous devez mettre à jour votre branche (le cas échéant) avant de résoudre le conflit.
[/list]

À noter : ce cas est courant si l'on modifie des fichiers [b]en.md[/b], les conflits sur les traductions sont rares, sauf si la personne n'avait pas vu qu'il y avait déjà une Pull Request, mais normalement nous vérifions que cela n'arrive pas, et corrigeons le problème le plus vite possible afin d'éviter ce cas de figure.

[b]Question :[/b] J'aimerais changer de branche, mais j'ai déjà des modifications en cours dessus, comment puis-je faire ?

[b]Réponse :[/b] Pour une raison ou une autre, vous souhaitez changer de branche, ou bien en créer une autre pour faire autre chose. Cependant, vous avez des modifications sur celle actuellement active.

> GitHub Desktop vous informera donc via [b][url=https://i.imgur.com/gouL9ie.png]cette fenêtre[/url][/b] si vous souhaitez laisser vos modifications sur cette branche ([b]Leave my changes on {nom de votre branche}[/b]), ce qui mettra les modifications sur cette branche en attente, et récupérables par la suite, ou si vous souhaitez que vos modifications sur la branche actuelle soient suivies sur la branche que vous avez sélectionnée ([b]Bring my changes to {nom de votre branche}[/b]).

Author: ```suggestion
GitHub Desktop vous informera donc via [b][url=https://imgur.com/gouL9ie.png]cette fenêtre[/url][/b] si vous souhaitez laisser vos modifications sur cette branche ([b]Leave my changes on {nom de votre branche}[/b]), ce qui mettra les modifications sur cette branche en attente, et récupérables par la suite, ou si vous souhaitez que vos modifications sur la branche actuelle soient suivies sur la branche que vous avez sélectionnée ([b]Bring my changes to {nom de votre branche}[/b]).
```

Selon ce que vous souhaitez faire, sélectionner l'une des deux options.

> Quand vous reviendrez sur votre branche en ayant sélectionné [b]Leave my changes on {nom de votre branche}[/b], vous avez un [b][url=https://i.imgur.com/QSYgQfe.png]bouton[/url][/b] qui se trouve dans la zone réservée habituellement a l'affichage des changements effectués.

Author: ```suggestion
Quand vous reviendrez sur votre branche en ayant sélectionné [b]Leave my changes on {nom de votre branche}[/b], vous avez un [b][url=https://imgur.com/QSYgQfe.png]bouton[/url][/b] qui se trouve dans la zone réservée habituellement a l'affichage des changements effectués.
```

> Cliquez sur ce bouton. Vous allez arriver [b][url=https://i.imgur.com/HL0h0ZM.png]ici[/url][/b]. Vous avez deux choix : [b]Restore[/b] pour restaurer les modifications qui s'affiche, [b]Discard[/b] pour abandonner les modifications affichées, elles seront totalement perdues après cela.

Author: ```suggestion
Cliquez sur ce bouton. Vous allez arriver [b][url=https://imgur.com/HL0h0ZM.png]ici[/url][/b]. Vous avez deux choix : [b]Restore[/b] pour restaurer les modifications qui s'affiche, [b]Discard[/b] pour abandonner les modifications affichées, elles seront totalement perdues après cela.
```

[b][size=150]Contribuer au projet de traduction de l’osu! wiki en français[/size][/b]

Bonjour à tous et à toutes !

J’aimerais vous présenter un projet qui a été créé le 3 août 2019 : un vaste projet de traduction de l’osu! wiki en français.

Ce projet a pour but de traduire l'[b][url=https://osu.ppy.sh/wiki/fr/Main_Page]osu! wiki[/url][/b] en français. Ce wiki est destiné à osu!, et réunit toutes les informations qui concernent le jeu.

Nous avons donc besoin de vous, pour construire un wiki compréhensible de toute la communauté française, pour vous, mais aussi et surtout pour les autres.

> Il n’y a pas forcément de prérequis pour rejoindre le projet : un niveau d’anglais et un orthographe corrects sont cependant indispensables pour comprendre les contenus en anglais et les restituer en français, si possible sans trop de fautes de français.

Reviewer: ```suggestion
Il n’y a pas forcément de prérequis pour rejoindre le projet : un niveau d’anglais et une orthographe corrects sont cependant indispensables pour comprendre les contenus en anglais et les restituer en français, si possible sans trop de fautes de français.
```

Reviewer: Je sais que c'est moi qui avais "corrigé" le `une` en `un` mais ma grosse fatigue au moment de ma review m'a fait faire des conneries, je vais bientôt essayer de faire une review complète sans fatigue histoire d'être sûr

Nous avons donc besoin de vous, pour construire un wiki compréhensible de toute la communauté française, pour vous, mais aussi et surtout pour les autres.
Il n’y a pas forcément de prérequis pour rejoindre le projet : un niveau d’anglais et une orthographe corrects sont cependant indispensables pour comprendre les contenus en anglais et les restituer en français, si possible sans trop de fautes de français.

Pour vous permettre de contribuer à tout cela facilement, nous avons décidé de réaliser un tutoriel afin de vous apprendre comment vous pouvez contribuer au mieux, correctement, et dans de bonnes conditions de travail.

Ce tutoriel a été écrit et imaginé par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], complété et amélioré par [b][url=https://osu.ppy.sh/users/10660738]Ganondorf[/url][/b].

[i]Si vous ne souhaitez pas traduire, vous pouvez tout de même participer au processus de review. Dans ce cas, dirigez vous directement dans la section [b]Proposer des modifications sur la Pull Request d'un autre traducteur[/b][/i]

Commençons !

> [b][size=150]GitHub[/size][/b]

Author: ```suggestion
[heading][centre][b][size=150]GitHub[/size][/b][/centre][/heading]
```

Toute la contribution du projet se passe sur [b]GitHub[/b], qui est une plateforme sur laquelle presque tous les projets ou logiciels sont basés, car elle permet à ces derniers d’être open-source (en clair, on peut voir la structure d’un logiciel gratuitement et sans risque, car ce seront les développeurs qui l’auront voulu !).

Pour commencer, nous allons vous montrer comment créer un compte sur cette plateforme, et comment mettre en place votre compte GitHub afin de pouvoir accueillir le projet.

> [notice][centre][b][size=150]Création du compte[/size][/b][/centre][/notice]

Author: ```suggestion
[heading][centre][b][size=150]Création du compte[/size][/b][/centre][/heading]
```

Avant de pouvoir traduire votre première page, vous devez faire une copie du projet que vous pourrez modifier sans risque.

Pour ce faire, vous allez repérer les trois boutons en haut à droite de votre page, et cliquer sur [b][url=https://imgur.com/IywtwKK.png]celui-ci[/url][/b]. [i]Fork[/i] (nom barbare en premier lieu) est un terme qui signifie que le projet sera copié (cloné en quelque sorte), en une version modifiable sans risque.
À partir de ce moment, le projet va être cloné. Cela peut prendre du temps : n’ayez crainte ! Cela est tout à fait normal, le projet est assez volumineux et prend donc un certain temps à se copier.

Une fois le projet copié sur votre compte, cela devrait ressembler à [b][url=https://imgur.com/7lzNZmV.png]ceci[/url][/b]. En premier lieu, vous allez vous dire que rien n’a changé, mais en réalité le projet est désormais modifiable sur un dépôt qui vous est propre, comme si vous en étiez le propriétaire alors qu’il ne s'agit que d'une simple copie de ce projet.
Vous pouvez voir la petite différence avec la page d'avant et celle ci [b][url=https://imgur.com/zEDdgR9.png]ici[/url][/b].

> [notice][centre][b][size=150]GitHub Desktop[/size][/b][/centre][/notice]

Author: ```suggestion
[heading][centre][b][size=150]GitHub Desktop[/size][/b][/centre][/heading]
```

Si tout va bien, le projet devrait se télécharger sur votre ordinateur. Cela sera également long, car le projet est volumineux.

Lorsque le projet aura été copié sur votre ordinateur, vous devriez avoir une fenêtre qui ressemble à [b][url=https://imgur.com/Sz5CYFB.png]celle-ci[/url][/b]. Cette fenêtre va vous demander comment vous comptez utiliser votre fork.
[list]
[*][b]To contribute to the parent project[/b] pour contribuer sur le projet [i]parent[/i] il s'agira dans notre cas de [b]ppy/osu-wiki[/b]
[*][b]For my own purposes[/b] pour contribuer sur votre propre fork, ce qui est la plupart du temps déconseillé pour les projets publics.
[/list]
Puisque nous voulons partager nos traductions, je vous conseille de cliquer sur [b]To contribute to the parent project[/b].

Après toutes ces étapes, vous y êtes ! voici [b][url=https://imgur.com/nPPkZp7.png]la fenêtre principale de GitHub Desktop[/url][/b].

Il vous reste encore un peu de chemin pour commencer votre contribution, mais n'ayez crainte ! Nous allons tout vous expliquer, encore un effort !

Pour le moment, nous allons laisser de côté GitHub Desktop, et nous allons commencer la mise en place de l’éditeur de texte.

> [notice][centre][b][size=150]Visual Studio Code[/size][/b][/centre][/notice]

Author: ```suggestion
[heading][centre][b][size=150]Visual Studio Code[/size][/b][/centre][/heading]
```

[list]
[*]GitHub a mis en place un système de [b]branches[/b] (branch en anglais), permettant d’inclure des modifications sur plusieurs de ces branches, ce qui vous permet de traduire plusieurs fichiers différents, sans que cela puisse vous gêner avec la précédente.
Pour créer une branche, vous devez aller dans GitHub Desktop, et cliquer sur [b][url=https://imgur.com/t7i3OFZ.png]ce bouton[/url][/b], qui va vous ouvrir [b][url=https://imgur.com/zvgusd1.png]une liste déroulante comme celle-ci[/url][/b] avec toutes les branches que vous possédez (vous n’en aurez aucune au départ).

Pour en créer une, cliquez sur [b]New branch[/b] en haut à droite, mettez ensuite le nom de votre branche (nous vous conseillons de mettre le nom de la page que vous souhaitez traduire (en remplaçant les espaces par des [b]-[/b] comme le nom ne doit pas contenir d’espaces, sinon il le fera pour vous), avec juste après la mention [b]FR[/b] pour signifier que c’est une traduction en langue française.
Lorsque vous aurez nommé votre branche, cliquez sur le bouton [b]Create branch[/b], et le tour est joué !

> [*]Tout les articles n’ont pas forcément besoin d’être traduits/mis à jour, c’est pour cela qu’un modérateur du projet a décidé de mettre en place [b][url=https://clayton.cc/osu-wiki-status/fr]ce site[/url][/b] qui recense toutes les pages nécessitant notre attention par catégorie. Vous devez éviter de traduire les pages qui se trouvent dans les catégories [b]Missing articles (outdated in English)[/b], ou toutes celles qui possèdent [b][url=https://imgur.com/6KSQCmB.png]ceci[/url][/b] et/ou [b][url=https://imgur.com/46EmsAQ.png]cela[/url][/b] sur le fichier [b]en.md[/b] (qui est le fichier en langue anglaise du fichier, celui sur lequel vous allez vous baser pour traduire)

Reviewer 2: ```suggestion
[*]Tous les articles n’ont pas forcément besoin d’être traduits/mis à jour, c’est pour cela qu’un modérateur du projet a décidé de mettre en place [b][url=https://clayton.cc/osu-wiki-status/fr]ce site[/url][/b] qui recense toutes les pages nécessitant notre attention par catégorie. Vous devez éviter de traduire les pages qui se trouvent dans les catégories [b]Missing articles (outdated in English)[/b], ou toutes celles qui possèdent [b][url=https://imgur.com/6KSQCmB.png]ceci[/url][/b] et/ou [b][url=https://imgur.com/46EmsAQ.png]cela[/url][/b] sur le fichier [b]en.md[/b] (qui est le fichier en langue anglaise du fichier, celui sur lequel vous allez vous baser pour traduire)
```

Le tag [b]outdated: true[/b] signifie que l’article n’est plus à jour, et qu'il peut à tout moment être modifié en ajoutant de nouvelles informations.
Le tag [b]need_cleanup: true[/b] signifie que l’article a besoin de nettoyage, soit d’informations superflues, soit au niveau de la mise en forme/écriture de l’article.

> En complément de cela, vous avez [b][url=https://github.com/ppy/osu-wiki/issues/2486]ces explications[/url][/b] sur ce qui a été dit.

Reviewer 2: Il manque pas quelque chose dans cette phrase ? Genre "...vous avez ici ces explications..." voire "les explications"

> En complément de cela, vous avez [b][url=https://github.com/ppy/osu-wiki/issues/2486]ces explications[/url][/b] sur ce qui a été dit.

Author: ```suggestion
En complément de cela, vous avez [b][url=https://github.com/ppy/osu-wiki/issues/2486]ces explications[/url][/b] sur ce qui a été dit plus haut.
```

[*]Avant de choisir votre page, vous devez vous assurer que cette dernière n’est pas déjà en train d’être traduite par un autre contributeur. C’est pour cela que nous avons mis en place [b][url=https://docs.google.com/spreadsheets/d/1zppXOH3PK32c0FwR8TC8QixxzoliEPcWNOjMfAfAv8o/edit#gid=1581693581]cette spreadsheet[/url][/b] qui liste les articles qui sont en cours de traduction, ceux qui sont en attente de vérification par d’autres traducteurs, ainsi que la liste (informative) des traductions terminées et publiées sur le wiki.

[*]Quand vous aurez choisi votre page, vous pouvez tout traduire, sauf :

Ce qui se trouve entre parenthèses avec un lien comme [b][url=https://imgur.com/GUmoQ38.png]celui-ci[/url][/b]. Ce sont des liens de redirections vers d’autres articles, vous ne devez pas y toucher, sinon la redirection ne fonctionnera plus.

Les termes spécifiques au jeu qui n’ont pas de traduction française (se référer à [b][url=https://gist.github.com/DoryanR/59b20e0fb40f27a371761f2fd002cccc]cette liste[/url][/b] si vous avez un doute).

> Sachez que votre traduction ne sera pas tout de suite mise sur le site : elle devra d'abord être vérifiée par d’autres traducteurs français avant qu’elle soit mise sur le site, n’ayez pas peur de faire des erreurs ! Nous sommes tous passer par la au début, nous savons que c'est dure au début.

Reviewer 2: Pas sûr pour le "e" de "dure" et pour le rajout du "ne" mais je trouve que ça rends plus fluide avec ce dernier.
```suggestion
Sachez que votre traduction ne sera pas tout de suite mise sur le site : elle devra d'abord être vérifiée par d’autres traducteurs français avant qu’elle ne soit mise sur le site, n’ayez pas peur de faire des erreurs ! Nous sommes tous passés par là au début, nous savons que c'est dur au début.
```

[*]Quand vous devez créer un fichier pour accueillir la traduction française d’une page, appelez votre fichier [b]fr.md[/b].

[*]Un conseil : quand vous souhaitez traduire une page, je vous conseille de prendre tout le contenu du fichier original en anglais, de le copier sur votre fichier de traduction, et de traduire directement dessus. Ça vous évite de devoir refaire toute la mise en forme, et c’est plus rapide.
[/list]
Maintenant que vous savez tout ça, vous pouvez vous lancer.

> [notice][centre][b][size=150]Mettre en ligne la contribution[/size][/b][/centre][/notice]

Reviewer 2: Je trouve que c'est plus sympa de mettre "votre"
```suggestion
[notice][centre][b][size=150]Mettre en ligne votre contribution[/size][/b][/centre][/notice]
```

> [notice][centre][b][size=150]Mettre en ligne votre contribution[/size][/b][/centre][/notice]

Author: ```suggestion
[heading][centre][b][size=150]Mettre en ligne votre contribution[/size][/b][/centre][/heading]
```

> Une fois votre traduction terminée, revenez sur GitHub Desktop, la fenêtre principale [b][url=https://imgur.com/QqLYndG.png]aura légèrement changé[/url][/b], du moment que vous avez sauvegarder votre fichier sur Visual Studio Code.

Reviewer 2: ```suggestion
Une fois votre traduction terminée, revenez sur GitHub Desktop, la fenêtre principale [b][url=https://imgur.com/QqLYndG.png]aura légèrement changée[/url][/b], du moment que vous avez sauvegardé votre fichier sur Visual Studio Code.
```

Juste en dessous, vous avez le titre de votre premier commit, qui est automatiquement transformé en votre titre pour votre Pull Request, avec la description en dessous (si vous en avez ajouté une).
Si vous allez en dessous, vous avez le même affichage que tout à l’heure sur GitHub Desktop, avec ce que vous avez ajouté au fichier (ou supprimé).

Une petite information importante : en bas de la zone de texte pour la description, vous avez [b][url=https://imgur.com/1ucKWCA.png]ceci[/url][/b]. Ce paramètre doit être coché, car il permet aux modérateurs du projet (ils sont appelés [b]Maintainers[/b]) d’accéder à votre branche et d’y apporter des modifications, comme ce que vous ferez plus tard. Cela n’est pas gênant de leur laisser l’accès à votre branche, car seuls eux pourront y accéder.

Lorsque vous aurez ajouté une description (vous pouvez simplement dire que vous avez traduit tel article, cela suffira), cliquez sur [b]Create pull request[/b].

Une fois votre Pull Request publiée, voici [b][url=https://imgur.com/VsIAK7L.png]à quoi cela ressemble[/url][/b].

> À partir de ce moment, vous avez fait le plus dur. Vous pouvez désormais attendre que d’autres traducteurs français regardent votre Pull Request, y apportent des suggestions, et que les modifications soient approuvées, vous pouvez pendant ce temps commencer à traduire un autre article.

Author: ```suggestion
À partir de ce moment, vous avez fait le plus dur. Vous pouvez désormais attendre que d’autres traducteurs français regardent votre Pull Request, y apportent des suggestions, et que les modifications soient approuvées, vous pouvez pendant ce temps commencer à traduire un autre article, ou faire des [i]reviews[/i] sur d'autres Pull Request.
```

> [notice][centre][b][size=150]Apporter des modifications à ma Pull Request[/size][/b][/centre][/notice]

Author: ```suggestion
[heading][centre][b][size=150]Apporter des modifications à ma Pull Request[/size][/b][/centre][/heading]
```

[*][b]Add suggestion to batch[/b] règlera votre problème si vous avez plusieurs suggestions, car il va prendre en compte que vous avez sélectionné cette modification, et quand vous en sélectionnez une autre, il la mettra aussi en attente.
Au final, vous aurez vos deux suggestions en un seul commit, plutôt qu'en deux séparé. L'avantage, c'est que cela rend votre Pull Request propre ; l'inconvénient : y'en a pas !
[/list]
Ensuite, quand vous aurez cliqué sur ce bouton, il sera remplacé par le précédent, avec un nombre à côté : il correspond au nombre de suggestions qui vont être appliquées sur votre fichier. Suivez ensuite les étapes pour poster votre commit.

Ces solutions évitent de repasser par Visual Studio Code, de chercher la ligne ou on vous propose une modification, appliquer sa modification, sauvegarder, faire un commit.
Cela prend un temps considérable. Avec ces solutions, vous gagnez un temps considérable dans le processus de review, qui peut être long selon le volume du fichier.

> [notice][centre][b][size=150]Proposer des modifications sur la Pull Request d'un autre contributeur[/size][/b][/centre][/notice]

Author: ```suggestion
[heading][centre][b][size=150]Proposer des modifications sur la Pull Request d'un autre contributeur[/size][/b][/centre][/heading]
```

[list]
[*]Un bouton [b]Add single comment[/b] qui va directement poster la suggestion
[*]Un bouton [b]Start a review[/b] qui va mettre votre suggestion en attente, et vous permettre de proposer d’autres modifications si vous en avez décelé plusieurs (conseillé car cela est mieux de poster toutes les suggestions d'un seul coup que de les poster une par une).
[/list]
Si vous avez terminé votre review avec la fonction [b]Start a review[/b], retournez en haut de la page, et cliquez sur [b][url=https://imgur.com/BafVZ2D.png]ce bouton[/url][/b]. Sélectionnez alors la case [b]Request changes[/b] pour indiquer que votre commentaire porte sur la suggestion de modifications sur la Pull Request.

Si vous estimez que la Pull Request est correcte, au lieu de sélectionner [b]Request changes[/b], cliquez sur [b]Approve[/b] pour approuver les modifications proposées par ce traducteur.

Nous avons fait le tour de tout ce que vous devez savoir sur le processus de traduction. Cela a été un peu long, mais il est important de détailler tout le processus dans son ensemble, afin que vous puissiez comprendre autant que possible ce processus qui reste complexe sans des explications claires et précises.

> [notice][centre][b][size=150]FAQ[/size][/b][/centre][/notice]

Author: ```suggestion
[heading][centre][b][size=150]FAQ[/size][/b][/centre][/heading]
```

GitHub Desktop vous informera donc via [b][url=https://imgur.com/gouL9ie.png]cette fenêtre[/url][/b] si vous souhaitez laisser vos modifications sur cette branche ([b]Leave my changes on {nom de votre branche}[/b]), ce qui mettra les modifications sur cette branche en attente, et récupérables par la suite, ou si vous souhaitez que vos modifications sur la branche actuelle soient suivies sur la branche que vous avez sélectionnée ([b]Bring my changes to {nom de votre branche}[/b]).
Selon ce que vous souhaitez faire, sélectionner l'une des deux options.

Quand vous reviendrez sur votre branche en ayant sélectionné [b]Leave my changes on {nom de votre branche}[/b], vous avez un [b][url=https://imgur.com/QSYgQfe.png]bouton[/url][/b] qui se trouve dans la zone réservée habituellement a l'affichage des changements effectués.
Cliquez sur ce bouton. Vous allez arriver [b][url=https://imgur.com/HL0h0ZM.png]ici[/url][/b]. Vous avez deux choix : [b]Restore[/b] pour restaurer les modifications qui s'affiche, [b]Discard[/b] pour abandonner les modifications affichées, elles seront totalement perdues après cela.

> [b]Question :[/b] Ma Pull Request à une croix rouge quand je regarde la liste des Pull Requests, qu'est ce que cela signifie, et comment la retirer ?

Reviewer 2: Petit oubli de saut de ligne je pense
```suggestion

[b]Question :[/b] Ma Pull Request à une croix rouge quand je regarde la liste des Pull Requests, qu'est ce que cela signifie, et comment la retirer ?
```

[b]Réponse :[/b] Cette croix correspond a une erreur présente dans les fichiers que vous avez soumis sur votre Pull Request. Ces erreurs correspondent la plupart du temps a un problème dans la mise en forme de l'article.
Cette erreur correspond à une intégration mise en place pour harmoniser les fichiers entre eux, et s'appelle [b]osu-wiki continuous integration[/b].

[b][/url=https://imgur.com/EnwlUoo.png]ici[/url][/b] vous avez un commit qui possède cette erreur.
Cliquez sur la croix, et vous verrez s'afficher [b][url=https://imgur.com/JqfyrbH.png]ceci[/url][/b], cliquez sur [b]Details[/b] pour l'intégration [b]osu-wiki continuous integration[/b].
Vous arrivez sur [b][/url=https://imgur.com/z9RFvcQ.png]cette écran[/url][/b] qui va vous expliquer ce qui ne va pas avec vos fichiers. Ce qui nous intéresse est ce qui ce trouve en couleur, avec un chemin d'accès : c'est la ou ce trouve votre fichier.

> Prenons par exemple la première ligne ; de gauche à droite, vous avez : La localisation du fichier, la ligne ou ce situe le problème, la colonne (la localisation de la lettre par rapport a la position de la marge), le nom de l'erreur, sa description, ainsi que la visualisation (en dessous) de la ligne en cause.

Reviewer 2: ```suggestion
Prenons par exemple la première ligne ; de gauche à droite, vous avez : La localisation du fichier, la ligne ou se situe le problème, la colonne (la localisation de la lettre par rapport a la position de la marge), le nom de l'erreur, sa description, ainsi que la visualisation (en dessous) de la ligne en cause.
```

Pour identifier l'erreur qui correspond à votre cas, dirigez vous [b][url=https://github.com/remarkjs/remark-lint/tree/main/packages]ici[/url][/b], cherchez ensuite le dossier qui correspond au nom de l'erreur que vous avez.
Vous aurez ensuite toutes les explications sur pourquoi elle apparaît, et les bonnes façons de ne pas l'avoir.

> [b][size=150]Liens utiles[/size][/b]

Author: ```suggestion
[notice][centre][b][size=150]Liens utiles[/size][/b][/centre][/notice]
```

[notice][centre][b][size=150]Contribuer au projet de traduction de l’osu! wiki en français[/size][/b][/centre][/notice]

Bonjour à tous et à toutes !

J’aimerais vous présenter un projet qui a été créé le 3 août 2019 : un vaste projet de traduction de l’osu! wiki en français.

Ce projet a pour but de traduire l'[b][url=https://osu.ppy.sh/wiki/fr/Main_Page]osu! wiki[/url][/b] en français. Ce wiki est destiné à osu!, et réunit toutes les informations qui concernent le jeu.

Nous avons donc besoin de vous, pour construire un wiki compréhensible de toute la communauté française, pour vous, mais aussi et surtout pour les autres.
Il n’y a pas forcément de prérequis pour rejoindre le projet : un niveau d’anglais et une orthographe corrects sont cependant indispensables pour comprendre les contenus en anglais et les restituer en français, si possible sans trop de fautes de français.

Pour vous permettre de contribuer à tout cela facilement, nous avons décidé de réaliser un tutoriel afin de vous apprendre comment vous pouvez contribuer au mieux, correctement, et dans de bonnes conditions de travail.

> Ce tutoriel a été écrit et imaginé par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], complété et amélioré par [b][url=https://osu.ppy.sh/users/10660738]Ganondorf[/url][/b].

Author: ```suggestion
Ce tutoriel a été écrit et imaginé par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], complété et amélioré par [b][url=https://osu.ppy.sh/users/10660738]Ganondorf[/url][/b] et [b][url=https://osu.ppy.sh/users/9066390]Satsukel[/url][/b].
```

Pour utiliser le logiciel dans le cadre de la contribution à l'osu! wiki, vous devez installer des extensions, permettant d’ajouter des fonctionnalités à l’éditeur.
Pour installer des extensions, vous devez cliquer sur [b][url=https://imgur.com/kCgmJ2l.png]ce bouton[/url][/b] pour accéder au menu des extensions.

Vous allez avoir une barre latérale sur votre gauche comme [b][url=https://imgur.com/I4Mc8Br.png]celle-ci[/url][/b].
En haut de cette barre latérale, vous avez une barre de recherche, sur cette dernière, vous allez installer deux extensions, indispensables pour la contribution : [b][url=https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens]GitLens — Git supercharged[/url][/b] de [b]Eric Amodio[/b] et [b][url=https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one]Markdown All in One[/url][/b] de [b]Yu Zhang[/b].
[list]
[*]GitLens est un outil permettant d’inclure toutes les fonctions de GitHub et de Git (voyez Git comme un [i]plugin[/i]) dans Visual Studio Code.

> [*]Markdown All in One est un outil permettant d'inclure toutes les fonctions du langage [b][url=https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet]Markdown[/url][/b] dans Visual Studio Code. Markdown se rapproche des langages [b][url]https://www.mediawiki.org/wiki/Manual:What_is_MediaWiki%3F/fr]MediaWiki[/url][/b] ou [b][url=https://www.phpbb-fr.com/forums/help/bbcode]BBCode[/url][/b], mais avec plus de possibilités et de simplicité.

Author: ```suggestion
[*]Markdown All in One est un outil permettant d'inclure toutes les fonctions du langage [b][url=https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet]Markdown[/url][/b] dans Visual Studio Code. Markdown se rapproche des langages [b][url=https://www.mediawiki.org/wiki/Manual:What_is_MediaWiki%3F/fr]MediaWiki[/url][/b] ou [b][url=https://www.phpbb-fr.com/forums/help/bbcode]BBCode[/url][/b], mais avec plus de possibilités et de simplicité.
```

[b][/url=https://imgur.com/EnwlUoo.png]ici[/url][/b] vous avez un commit qui possède cette erreur.
Cliquez sur la croix, et vous verrez s'afficher [b][url=https://imgur.com/JqfyrbH.png]ceci[/url][/b], cliquez sur [b]Details[/b] pour l'intégration [b]osu-wiki continuous integration[/b].
Vous arrivez sur [b][/url=https://imgur.com/z9RFvcQ.png]cette écran[/url][/b] qui va vous expliquer ce qui ne va pas avec vos fichiers. Ce qui nous intéresse est ce qui ce trouve en couleur, avec un chemin d'accès : c'est la ou ce trouve votre fichier.

Prenons par exemple la première ligne ; de gauche à droite, vous avez : La localisation du fichier, la ligne ou se situe le problème, la colonne (la localisation de la lettre par rapport a la position de la marge), le nom de l'erreur, sa description, ainsi que la visualisation (en dessous) de la ligne en cause.

Pour identifier l'erreur qui correspond à votre cas, dirigez vous [b][url=https://github.com/remarkjs/remark-lint/tree/main/packages]ici[/url][/b], cherchez ensuite le dossier qui correspond au nom de l'erreur que vous avez.
Vous aurez ensuite toutes les explications sur pourquoi elle apparaît, et les bonnes façons de ne pas l'avoir.

> [notice][centre][b][size=150]Liens utiles[/size][/b][/centre][/notice]

Author: ```suggestion
[heading][centre][b][size=150]Liens utiles[/size][/b][/centre][/heading]
```

> 

Author: ```suggestion

```

Bonjour à tous et à toutes !

J’aimerais vous présenter un projet qui a été créé le 3 août 2019 : un vaste projet de traduction de l’osu! wiki en français.

Ce projet a pour but de traduire l'[b][url=https://osu.ppy.sh/wiki/fr/Main_Page]osu! wiki[/url][/b] en français. Ce wiki est destiné à osu!, et réunit toutes les informations qui concernent le jeu.

Nous avons donc besoin de vous, pour construire un wiki compréhensible de toute la communauté française, pour vous, mais aussi et surtout pour les autres.
Il n’y a pas forcément de prérequis pour rejoindre le projet : un niveau d’anglais et une orthographe corrects sont cependant indispensables pour comprendre les contenus en anglais et les restituer en français, si possible sans trop de fautes de français.

Pour vous permettre de contribuer à tout cela facilement, nous avons décidé de réaliser un tutoriel afin de vous apprendre comment vous pouvez contribuer au mieux, correctement, et dans de bonnes conditions de travail.

Ce tutoriel a été écrit et imaginé par [b][url=https://osu.ppy.sh/users/9807004]Doryan[/url][/b], complété et amélioré par [b][url=https://osu.ppy.sh/users/10660738]Ganondorf[/url][/b].

> [i]Si vous ne souhaitez pas traduire, vous pouvez tout de même participer au processus de review. Dans ce cas, dirigez vous directement dans la section [b]Proposer des modifications sur la Pull Request d'un autre traducteur[/b][/i]
> Commençons !

Author: ```suggestion
[i]Si vous ne souhaitez pas traduire, vous pouvez tout de même participer au processus de review. Dans ce cas, dirigez vous directement dans la section [b]Proposer des modifications sur la Pull Request d'un autre traducteur[/b][/i]

Commençons !
```

Si rien ne s'affiche, cliquez sur le bouton [b]Clone a repository from the Internet…[/b], [b][url=https://imgur.com/GVtkqIZ.png]une petite fenêtre va s’ouvrir[/url][/b], si vous ne voyez toujours rien, allez dans la section [b]URL[/b], et tapez [b]{votre nom d’utilisateur}/osu-wiki[/b].
Si tout va bien, le projet devrait se télécharger sur votre ordinateur. Cela sera également long, car le projet est volumineux.

Lorsque le projet aura été copié sur votre ordinateur, vous devriez avoir une fenêtre qui ressemble à [b][url=https://imgur.com/Sz5CYFB.png]celle-ci[/url][/b]. Cette fenêtre va vous demander comment vous comptez utiliser votre fork.
[list]
[*][b]To contribute to the parent project[/b] pour contribuer sur le projet [i]parent[/i] il s'agira dans notre cas de [b]ppy/osu-wiki[/b]
[*][b]For my own purposes[/b] pour contribuer sur votre propre fork, ce qui est la plupart du temps déconseillé pour les projets publics.
[/list]

> 

Author: ```suggestion
```

Pour utiliser le logiciel dans le cadre de la contribution à l'osu! wiki, vous devez installer des extensions, permettant d’ajouter des fonctionnalités à l’éditeur.
Pour installer des extensions, vous devez cliquer sur [b][url=https://imgur.com/kCgmJ2l.png]ce bouton[/url][/b] pour accéder au menu des extensions.

Vous allez avoir une barre latérale sur votre gauche comme [b][url=https://imgur.com/I4Mc8Br.png]celle-ci[/url][/b].
En haut de cette barre latérale, vous avez une barre de recherche, sur cette dernière, vous allez installer deux extensions, indispensables pour la contribution : [b][url=https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens]GitLens — Git supercharged[/url][/b] de [b]Eric Amodio[/b] et [b][url=https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one]Markdown All in One[/url][/b] de [b]Yu Zhang[/b].
[list]
[*]GitLens est un outil permettant d’inclure toutes les fonctions de GitHub et de Git (voyez Git comme un [i]plugin[/i]) dans Visual Studio Code.

> [*]Markdown All in One est un outil permettant d'inclure toutes les fonctions du langage [b][url=https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet]Markdown[/url][/b] dans Visual Studio Code. Markdown se rapproche des langages [b][/url]https://www.mediawiki.org/wiki/Manual:What_is_MediaWiki%3F/fr]MediaWiki[/url][/b] ou [b][url=https://www.phpbb-fr.com/forums/help/bbcode]BBCode[/url][/b], mais avec plus de possibilités et de simplicité.

Author: ```suggestion
[*]Markdown All in One est un outil permettant d'inclure toutes les fonctions du langage [b][url=https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet]Markdown[/url][/b] dans Visual Studio Code. Markdown se rapproche des langages [b][url]https://www.mediawiki.org/wiki/Manual:What_is_MediaWiki%3F/fr]MediaWiki[/url][/b] ou [b][url=https://www.phpbb-fr.com/forums/help/bbcode]BBCode[/url][/b], mais avec plus de possibilités et de simplicité.
```

[/list]

> 

Author: ```suggestion
```

Ce qui se trouve entre parenthèses avec un lien comme [b][url=https://imgur.com/GUmoQ38.png]celui-ci[/url][/b]. Ce sont des liens de redirections vers d’autres articles, vous ne devez pas y toucher, sinon la redirection ne fonctionnera plus.

Les termes spécifiques au jeu qui n’ont pas de traduction française (se référer à [b][url=https://gist.github.com/DoryanR/59b20e0fb40f27a371761f2fd002cccc]cette liste[/url][/b] si vous avez un doute).

Sachez que votre traduction ne sera pas tout de suite mise sur le site : elle devra d'abord être vérifiée par d’autres traducteurs français avant qu’elle soit mise sur le site, n’ayez pas peur de faire des erreurs ! Nous sommes tous passer par la au début, nous savons que c'est dure au début.

[*]Quand vous devez créer un fichier pour accueillir la traduction française d’une page, appelez votre fichier [b]fr.md[/b].

[*]Un conseil : quand vous souhaitez traduire une page, je vous conseille de prendre tout le contenu du fichier original en anglais, de le copier sur votre fichier de traduction, et de traduire directement dessus. Ça vous évite de devoir refaire toute la mise en forme, et c’est plus rapide.
[/list]

> 

Author: ```suggestion
```

Maintenant que vous savez tout ça, vous pouvez vous lancer.

[b][size=150]Mettre en ligne la contribution[/size][/b]

Une fois votre traduction terminée, revenez sur GitHub Desktop, la fenêtre principale [b][url=https://imgur.com/QqLYndG.png]aura légèrement changé[/url][/b], du moment que vous avez sauvegarder votre fichier sur Visual Studio Code.

> Sur la gauche, vous avez la liste des fichiers que vous avez créés, supprimés et modifiés. Un [b]+[/b] [color=#008000]vert[/color] apparaîtra à côté du nom du fichier s'il s'agit d'un fichier créé, un [b].[/b] [color=#FFA500]orange[/color] s'il s'agit d'un fichier modifié, et un [b]-[/b] [color=#FFA500]rouge[/color] s'il s'agit d'un fichier supprimé.

Author: ```suggestion
Sur la gauche, vous avez la liste des fichiers que vous avez créés, supprimés et modifiés. Un [b]+[/b] [color=#008000]vert[/color] apparaîtra à côté du nom du fichier s'il s'agit d'un fichier créé, un [b].[/b] [color=#FFA500]orange[/color] s'il s'agit d'un fichier modifié, et un [b]-[/b] [color=#FF0000]rouge[/color] s'il s'agit d'un fichier supprimé.
```

Pour les résoudre : (réalisé à l'aide du guide du contributeur sur l'osu! wiki, nous avons besoin d'un cas concret d'un contributeur pour réaliser une procédure plus explicite)
[list]
[*] Cliquez sur le bouton [b]Resolve conflicts[/b]
[*] Vous verrez des bouts de texte qui sont entourés avec [b]<<<<<<<[/b] et [b]=======[/b] : ce sont vos modifications
[*] Vous verrez des bouts de texte qui sont entourés avec [b]=======[/b] et [b]>>>>>>> master[/b] : ce sont les modifications qui sont en conflit avec les vôtres
[*] Pour résoudre le conflit, retirez tous les marqueurs présents en surbrillance, en prenant garde que les modifications [i]"entrantes"[/i] ne soient pas en double avec celles déjà présentes ([i]"sortantes"[/i] donc), sinon, retirez la partie que vous aviez réalisé avant votre conflit (si les modifications en conflit sont le but de votre Pull Request, conservez-les)
[*] Si tous les conflits sont résolus, vous pouvez de nouveau cliquer sur le bouton [b]Resolve conflicts[/b].

Si ce bouton est grisé, vous devez mettre à jour votre branche (le cas échéant) avant de résoudre le conflit.
[/list]

> 

Author: ```suggestion
```

[b][/url=https://imgur.com/EnwlUoo.png]ici[/url][/b] vous avez un commit qui possède cette erreur.
Cliquez sur la croix, et vous verrez s'afficher [b][url=https://imgur.com/JqfyrbH.png]ceci[/url][/b], cliquez sur [b]Details[/b] pour l'intégration [b]osu-wiki continuous integration[/b].
Vous arrivez sur [b][/url=https://imgur.com/z9RFvcQ.png]cette écran[/url][/b] qui va vous expliquer ce qui ne va pas avec vos fichiers. Ce qui nous intéresse est ce qui ce trouve en couleur, avec un chemin d'accès : c'est la ou ce trouve votre fichier.

Prenons par exemple la première ligne ; de gauche à droite, vous avez : La localisation du fichier, la ligne ou ce situe le problème, la colonne (la localisation de la lettre par rapport a la position de la marge), le nom de l'erreur, sa description, ainsi que la visualisation (en dessous) de la ligne en cause.

Pour identifier l'erreur qui correspond à votre cas, dirigez vous [b][url=https://github.com/remarkjs/remark-lint/tree/main/packages]ici[/url][/b], cherchez ensuite le dossier qui correspond au nom de l'erreur que vous avez.

> Vous aurez ensuite toutes les explications sur pourquoi elle apparaît, et les bonnes façons de ne pas l'avoir.
> [b][size=150]Liens utiles[/size][/b]

Author: ```suggestion
Vous aurez ensuite toutes les explications sur pourquoi elle apparaît, et les bonnes façons de ne pas l'avoir.

[b][size=150]Liens utiles[/size][/b]
```

[notice][centre][b][size=150]Contribuer au projet de traduction de l’osu! wiki en français[/size][/b][/centre][/notice]

Bonjour à tous et à toutes !

J’aimerais vous présenter un projet qui a été créé le 3 août 2019 : un vaste projet de traduction de l’osu! wiki en français.

Ce projet a pour but de traduire l'[b][url=https://osu.ppy.sh/wiki/fr/Main_Page]osu! wiki[/url][/b] en français. Ce wiki est destiné à osu!, et réunit toutes les informations qui concernent le jeu.

> Nous avons donc besoin de vous, pour construire un wiki compréhensible de toute la communauté française, pour vous, mais aussi et surtout pour les autres.

Reviewer 3: ```suggestion
Nous avons donc besoin de vous, pour construire un wiki compris par toute la communauté française, pour vous, mais aussi et surtout pour les autres.
```

[i]Si vous ne souhaitez pas traduire, vous pouvez tout de même participer au processus de review. Dans ce cas, dirigez vous directement dans la section [b]Proposer des modifications sur la Pull Request d'un autre traducteur[/b][/i]

Commençons !

[heading][centre][b][size=150]GitHub[/size][/b][/centre][/heading]

Toute la contribution du projet se passe sur [b]GitHub[/b], qui est une plateforme sur laquelle presque tous les projets ou logiciels sont basés, car elle permet à ces derniers d’être open-source (en clair, on peut voir la structure d’un logiciel gratuitement et sans risque, car ce seront les développeurs qui l’auront voulu !).

Pour commencer, nous allons vous montrer comment créer un compte sur cette plateforme, et comment mettre en place votre compte GitHub afin de pouvoir accueillir le projet.

[heading][centre][b][size=150]Création du compte[/size][/b][/centre][/heading]

Pour commencer, nous allons créer un compte. Pour ce faire, dirigez-vous [b][url=https://github.com/signup?ref_cta=Sign+up&ref_loc=header+logged+out&ref_page=%2F&source=header-home]ici[/url][/b], et laissez-vous guider.

> Quand votre compte est configuré (et que vous l’aurez peut-être paramétré à votre goût), vous allez pouvoir accéder au [i]dépôt[/i] du projet. 

Reviewer 3: ```suggestion
Quand votre compte est configuré (et qu'il sera paramétré à votre goût), vous allez pouvoir accéder au [i]dépôt[/i] du projet. 
```

À partir de ce moment, le projet va être cloné. Cela peut prendre du temps : n’ayez crainte ! Cela est tout à fait normal, le projet est assez volumineux et prend donc un certain temps à se copier.

Une fois le projet copié sur votre compte, cela devrait ressembler à [b][url=https://imgur.com/7lzNZmV.png]ceci[/url][/b]. En premier lieu, vous allez vous dire que rien n’a changé, mais en réalité le projet est désormais modifiable sur un dépôt qui vous est propre, comme si vous en étiez le propriétaire alors qu’il ne s'agit que d'une simple copie de ce projet.
Vous pouvez voir la petite différence avec la page d'avant et celle ci [b][url=https://imgur.com/zEDdgR9.png]ici[/url][/b].

[heading][centre][b][size=150]GitHub Desktop[/size][/b][/centre][/heading]

Maintenant que vous avez cloné le projet, nous allons pouvoir parler d’un logiciel très important : [b]GitHub Desktop[/b].

> Qu’est-ce que GitHub Desktop ? Il s'agit d'un logiciel (créé par GitHub) totalement gratuit qui va vous permettre de lier GitHub avec un éditeur de texte (que nous verrons plus tard), afin de pouvoir déposer vos traductions sur le projet. Il est indispensable, car sans lui, ce processus serait compliqué !

Reviewer 3: ```suggestion
Qu’est-ce que GitHub Desktop ? Il s'agit d'un logiciel (créé par GitHub) totalement gratuit qui va vous permettre de lier GitHub avec un éditeur de texte (que nous verrons plus tard), afin de pouvoir déposer vos traductions sur le projet. Il est indispensable, car sans lui, ce processus serait compliqué ! Il est disponible sur Windows et MacOS.
```

Reviewer 3: J'ai jugé important de spécifier que le logiciel est dispo sur MacOS et sur Windows

Une fois téléchargé, il va vous proposer [b][url=https://imgur.com/jLQMOvt.png]de vous connecter[/url][/b], sélectionnez l’option [b]Sign in to GitHub.com[/b].

GitHub Desktop va alors vous proposer de [b][url=https://imgur.com/IiLGTEk.png]configurer votre adresse mail[/url][/b], je vous conseille de la laisser comme telle, à moins que vous vouliez laisser apparaître celle avec laquelle vous avez créé votre compte publiquement. Cliquez donc sur [b]Finish[/b].

Enfin, il va vous proposer [b][url=https://imgur.com/gVZPZyn.png]de cloner le dépôt sur votre ordinateur[/url][/b]. Si tout va bien, vous aurez à droite la liste des dépôts qui sont présents sur votre compte. Cliquez sur [b]{votre nom d’utilisateur}/osu-wiki[/b], puis sur [b]Clone {votre nom d’utilisateur}/osu-wiki[/b], et une nouvelle fois sur [b]Clone[/b].

Si rien ne s'affiche, cliquez sur le bouton [b]Clone a repository from the Internet…[/b], [b][url=https://imgur.com/GVtkqIZ.png]une petite fenêtre va s’ouvrir[/url][/b], si vous ne voyez toujours rien, allez dans la section [b]URL[/b], et tapez [b]{votre nom d’utilisateur}/osu-wiki[/b].

> Si tout va bien, le projet devrait se télécharger sur votre ordinateur. Cela sera également long, car le projet est volumineux.

Reviewer 3: ```suggestion
Si tout va bien, le projet se téléchargera sur votre ordinateur. Cela sera également long, car le projet est volumineux.
```

Reviewer 3: J'ai suivi la conjugaison par rapport au verbe "sera", donc j'ai mis "téléchargé" au même futur que le verbe être dans la phrase suivante

Visual Studio Code, c’est un logiciel de traitement de texte comme il en existe des milliers. Sauf que ce dernier est plus destiné à l’écriture de lignes de codes et au développement. C’est comme [b]Notepad++[/b] pour les connaisseurs, mais avec des possibilités presque infinies.
Comme GitHub Desktop, Visual Studio Code est gratuit, et est [i]normalement[/i] plus simple d’utilisation.

Pour commencer, [b][url=https://code.visualstudio.com/]installez le logiciel[/url][/b]. Installez-le comme un logiciel classique, tout comme GitHub Desktop.

Ensuite, quand tout est installé, vous devriez tomber sur une fenêtre comme [b][url=https://imgur.com/qvyqr7Y.png]celle-ci[/url][/b], c’est la fenêtre principale de Visual Studio Code.

Pour utiliser le logiciel dans le cadre de la contribution à l'osu! wiki, vous devez installer des extensions, permettant d’ajouter des fonctionnalités à l’éditeur.
Pour installer des extensions, vous devez cliquer sur [b][url=https://imgur.com/kCgmJ2l.png]ce bouton[/url][/b] pour accéder au menu des extensions.

Vous allez avoir une barre latérale sur votre gauche comme [b][url=https://imgur.com/I4Mc8Br.png]celle-ci[/url][/b].

> En haut de cette barre latérale, vous avez une barre de recherche, sur cette dernière, vous allez installer deux extensions, indispensables pour la contribution : [b][url=https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens]GitLens — Git supercharged[/url][/b] de [b]Eric Amodio[/b] et [b][url=https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one]Markdown All in One[/url][/b] de [b]Yu Zhang[/b].

Reviewer 3: ```suggestion
En haut de cette barre latérale, vous avez une barre de recherche, et sur cette dernière, vous allez installer deux extensions, indispensables pour la contribution : [b][url=https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens]GitLens — Git supercharged[/url][/b] de [b]Eric Amodio[/b] et [b][url=https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one]Markdown All in One[/url][/b] de [b]Yu Zhang[/b].
```

[*]Quand vous aurez choisi votre page, vous pouvez tout traduire/modifier, sauf :

Ce qui se trouve entre parenthèses avec un lien comme [b][url=https://imgur.com/GUmoQ38.png]celui-ci[/url][/b]. Ce sont des liens de redirections vers d’autres articles, vous ne devez pas y toucher, sinon la redirection ne fonctionnera plus.

Les termes spécifiques au jeu qui n’ont pas de traduction française (se référer à [b][url=https://gist.github.com/DoryanR/59b20e0fb40f27a371761f2fd002cccc]cette liste[/url][/b] si vous avez un doute).

Les [b][url=https://imgur.com/C7ECeDs.png]tags[/url][/b] qui ce trouve en haut de la page.
Dans ce genre de cas, vous devez conserver les tags actuels, et mettre la traduction de ces derniers en dessous.

> Les termes entre [i]<!-- -->[/i] comme [b][url=url=https://imgur.com/DTGk8bz.png]ici[/url][/b].

Author: ```suggestion
Les termes entre [i]<!-- -->[/i] comme [b][url=https://imgur.com/DTGk8bz.png]ici[/url][/b].
```

> Ces termes sont des notes qui n'apparaissent pas dans l'article, et qui permettent a ceux qui le modifie de les consulter. Ils contiennent la plupart du temps des listes de choses à faire (rajouts, suppressions, récriture, etc...).

Reviewer 3: ```suggestion
Ces termes sont des notes qui n'apparaissent pas dans l'article, et qui permettent à ceux qui le modifient de les consulter. Ils contiennent la plupart du temps des listes de choses à faire (rajouts, suppressions, récriture, etc...).
```

Dans le cadre de la contribution, cela peut être supprimer sans crainte, car elle pollue plus qu'autre chose la traduction, et n'a pas d'utilité, vu que cela concerne le fichier [b]en.md[/b].

> Sachez que votre traduction ne sera pas tout de suite mise sur le site : elle devra d'abord être vérifiée par d’autres traducteurs français avant qu’elle ne soit mise sur le site, n’ayez pas peur de faire des erreurs ! Nous sommes tous passés par là au début, nous savons que c'est dur au début.

Reviewer 3: ```suggestion
Sachez que votre traduction ne sera pas tout de suite mise sur le site : elle devra d'abord être vérifiée par d’autres traducteurs français avant qu’elle ne soit mise sur le site, n’ayez pas peur de faire des erreurs ! Nous sommes tous passés par là et nous savons que c'est dur au début.
```

Si vous choisissez de retourner sur le dépôt, vous pourrez observer un [b][url=https://imgur.com/1sSHcQZ.png]petit changement[/url][/b] en haut de votre écran. Cette fenêtre en orange est la branche que vous venez juste de publier sur GitHub, cliquez donc sur [b]Compare & pull request[/b].

Vous allez arriver sur [b][url=https://imgur.com/9ENAr8z.png]cette fenêtre[/url][/b]. C’est là que vous pouvez voir les informations sur votre branche, avec les modifications que vous voulez proposer.
Ce qui se trouve tout en haut, ne vous en occupez pas, ce sont simplement les paramètres qui définissent sur quel dépôt du projet il doit se baser pour déterminer la différence entre les deux versions du fichier.

Juste en dessous, vous avez le titre de votre premier commit, qui est automatiquement transformé en votre titre pour votre Pull Request, avec la description en dessous (si vous en avez ajouté une).
Si vous allez en dessous, vous avez le même affichage que tout à l’heure sur GitHub Desktop, avec ce que vous avez ajouté au fichier (ou supprimé).

> Une petite information importante : en bas de la zone de texte pour la description, vous avez [b][url=https://imgur.com/1ucKWCA.png]ceci[/url][/b]. Ce paramètre doit être coché, car il permet aux modérateurs du projet (ils sont appelés [b]Maintainers[/b]) d’accéder à votre branche et d’y apporter des modifications, comme ce que vous ferez plus tard. Cela n’est pas gênant de leur laisser l’accès à votre branche, car seuls eux pourront y accéder.

Reviewer 3: ```suggestion
Une petite information importante : en bas de la zone de texte pour la description, vous avez [b][url=https://imgur.com/1ucKWCA.png]ceci[/url][/b]. Ce paramètre doit être coché, car il permet aux modérateurs du projet (ils sont appelés [b]Maintainers[/b]) d’accéder à votre branche et d’y apporter des modifications, comme ce que vous ferez plus tard. Cela n’est pas gênant de leur laisser l’accès à votre branche, car eux seuls pourront y accéder.
```

Lorsque vous aurez ajouté une description (vous pouvez simplement dire que vous avez traduit tel article, cela suffira), cliquez sur [b]Create pull request[/b].

Une fois votre Pull Request publiée, voici [b][url=https://imgur.com/VsIAK7L.png]à quoi cela ressemble[/url][/b].

> À partir de ce moment, vous avez fait le plus dur. Vous pouvez désormais attendre que d’autres traducteurs français regardent votre Pull Request, y apportent des suggestions, et que les modifications soient approuvées, vous pouvez pendant ce temps commencer à traduire un autre article, ou faire des [i]reviews[/i] sur d'autres Pull Request.

Reviewer 3: ```suggestion
À partir de ce moment, vous avez fait le plus dur. Vous pouvez désormais attendre que d’autres traducteurs français regardent votre Pull Request, y apportent des suggestions, et que les modifications soient approuvées. Vous pouvez pendant ce temps commencer à traduire un autre article, ou faire des [i]reviews[/i] sur d'autres Pull Request.
```

[heading][centre][b][size=150]Apporter des modifications à ma Pull Request[/size][/b][/centre][/heading]

Des modifications ont été proposées sur votre Pull Request ? Voici comment les accepter, ou proposer des modifications si elles vous semblent incorrectes.

Si l’un des traducteurs propose des modifications sur votre Pull Request, vous devez soit appliquer les modifications que l’on vous a proposées, soit les commenter si vous n’êtes pas d’accord.
Dans les deux cas, tout se passe dans l’onglet [b]Files changed[/b].

L’onglet ressemble à [b][url=https://imgur.com/YIUOdib.png]ceci[/url][/b].

Pour faire un exemple, je vais simuler une demande de modification de ma propre Pull Request (c’est possible, et cela est plus simple que de repasser par Visual Studio Code pour apporter des modifications ; nous verrons cela un peu plus tard).

Imaginons que je souhaite mettre une majuscule au mot [b]beatmapping[/b], qui se trouve à la ligne 7.
[b][url=https://imgur.com/IQ7JDku.png]Voici a quoi ressemblera la suggestion[/url][/b].

> La partie rouge s'agit de ce qui se trouvait avant la proposition (ce qui est souligné est l’élément sur lequel la personne qui propose une modification agit, ici c’est [b]Beatmapping[/b].

Reviewer 3: ```suggestion
La partie rouge correspond à ce qui se trouvait avant la proposition (ce qui est souligné est l’élément sur lequel la personne qui propose une modification agit, ici c’est [b]Beatmapping[/b].
```

Reviewer 3: La façon dont est construite la phrase avec "s'agit de" est selon moi pas très claire.

[list]
[*][b]Commit suggestion[/b] va appliquer directement la suggestion, après avoir donné un nom à votre commit. Pratique quand il n’y a qu’une seule suggestion à proposer, mais qui pose problème quand vous en avez plusieurs (comme ici)
[*][b]Add suggestion to batch[/b] règlera votre problème si vous avez plusieurs suggestions, car il va prendre en compte que vous avez sélectionné cette modification, et quand vous en sélectionnerez une autre, il la mettra aussi en attente.
Au final, vous aurez vos deux suggestions en un seul commit, plutôt qu'en deux séparés. L'avantage, c'est que cela rend votre Pull Request plus propre ; l'inconvénient : y'en a pas !
[/list]
Ensuite, quand vous aurez cliqué sur ce bouton, il sera remplacé par le précédent, avec un nombre à côté : il correspond au nombre de suggestions qui vont être appliquées sur votre fichier. Suivez ensuite les étapes pour poster votre commit.

Ces solutions évitent de repasser par Visual Studio Code, de chercher la ligne où on vous propose une modification, appliquer sa modification, sauvegarder, et faire un commit.

> Cela prend un temps considérable. Avec ces solutions, vous gagnez un temps considérable dans le processus de review, qui peut être long selon le volume du fichier.

Reviewer 3: ```suggestion
Cela prend un temps considérable. Avec ces solutions, vous gagnez énormément de temps dans le processus de review, qui peut être long selon le volume du fichier.
```

Reviewer 3: La tournure est mieux comme ceci car la répétition de "considérable" rend la phrase très lourde.
